<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="Best practice recommendations for data packages, described by Ecological Metadata Language (EML). 2022-05-25 version">

<title>3&nbsp; Introduction – Data Package Best Practices</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../guide-domain-specific/preface.html" rel="next">
<link href="../guide-special-cases/preface.html" rel="prev">
<link href="../edi-logo.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../guide-special-cases/preface.html">Data Package Design for Special Cases</a></li><li class="breadcrumb-item"><a href="../guide-special-cases/dataset-design-special-cases.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Introduction</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Data Package Best Practices</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/ediorg/dataset_preparation_guides" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Overview</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../about.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About This Site</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../guide-eml-bp/preface.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Best Practices for Dataset Metadata in Ecological Metadata Language (EML)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../guide-eml-bp/introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../guide-eml-bp/content-recommendations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Content recommendations for Elements and Attributes</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../guide-special-cases/preface.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Package Design for Special Cases</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../guide-special-cases/dataset-design-special-cases.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../guide-domain-specific/preface.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scientific domain specific metadata guide</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../guide-domain-specific/scientific-domain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#code" id="toc-code" class="nav-link active" data-scroll-target="#code"><span class="header-section-number">3.1</span> Code</a>
  <ul class="collapse">
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction"><span class="header-section-number">3.1.1</span> Introduction</a></li>
  <li><a href="#recommendations-for-data-packages" id="toc-recommendations-for-data-packages" class="nav-link" data-scroll-target="#recommendations-for-data-packages"><span class="header-section-number">3.1.2</span> Recommendations for data packages</a></li>
  <li><a href="#resources" id="toc-resources" class="nav-link" data-scroll-target="#resources"><span class="header-section-number">3.1.3</span> Resources</a></li>
  </ul></li>
  <li><a href="#model-based-datasets" id="toc-model-based-datasets" class="nav-link" data-scroll-target="#model-based-datasets"><span class="header-section-number">3.2</span> Model-Based Datasets</a>
  <ul class="collapse">
  <li><a href="#introduction-1" id="toc-introduction-1" class="nav-link" data-scroll-target="#introduction-1"><span class="header-section-number">3.2.1</span> Introduction</a></li>
  <li><a href="#recommendations-for-data-packages-1" id="toc-recommendations-for-data-packages-1" class="nav-link" data-scroll-target="#recommendations-for-data-packages-1"><span class="header-section-number">3.2.2</span> Recommendations for data packages</a></li>
  <li><a href="#example-data-packages-in-edi" id="toc-example-data-packages-in-edi" class="nav-link" data-scroll-target="#example-data-packages-in-edi"><span class="header-section-number">3.2.3</span> Example data packages in EDI</a></li>
  <li><a href="#resources-1" id="toc-resources-1" class="nav-link" data-scroll-target="#resources-1"><span class="header-section-number">3.2.4</span> Resources</a></li>
  </ul></li>
  <li><a href="#images-and-documents-as-data" id="toc-images-and-documents-as-data" class="nav-link" data-scroll-target="#images-and-documents-as-data"><span class="header-section-number">3.3</span> Images and Documents as Data</a>
  <ul class="collapse">
  <li><a href="#introduction-2" id="toc-introduction-2" class="nav-link" data-scroll-target="#introduction-2"><span class="header-section-number">3.3.1</span> Introduction</a></li>
  <li><a href="#recommendations-for-data-packages-2" id="toc-recommendations-for-data-packages-2" class="nav-link" data-scroll-target="#recommendations-for-data-packages-2"><span class="header-section-number">3.3.2</span> Recommendations for data packages</a></li>
  <li><a href="#example-data-packages-in-edi-1" id="toc-example-data-packages-in-edi-1" class="nav-link" data-scroll-target="#example-data-packages-in-edi-1"><span class="header-section-number">3.3.3</span> Example data packages in EDI</a></li>
  <li><a href="#resources-2" id="toc-resources-2" class="nav-link" data-scroll-target="#resources-2"><span class="header-section-number">3.3.4</span> Resources</a></li>
  </ul></li>
  <li><a href="#spatial-data" id="toc-spatial-data" class="nav-link" data-scroll-target="#spatial-data"><span class="header-section-number">3.4</span> Spatial Data</a>
  <ul class="collapse">
  <li><a href="#introduction-3" id="toc-introduction-3" class="nav-link" data-scroll-target="#introduction-3"><span class="header-section-number">3.4.1</span> Introduction</a></li>
  <li><a href="#recommendations-for-data-packages-3" id="toc-recommendations-for-data-packages-3" class="nav-link" data-scroll-target="#recommendations-for-data-packages-3"><span class="header-section-number">3.4.2</span> Recommendations for data packages</a></li>
  </ul></li>
  <li><a href="#data-gathered-with-small-moving-platforms" id="toc-data-gathered-with-small-moving-platforms" class="nav-link" data-scroll-target="#data-gathered-with-small-moving-platforms"><span class="header-section-number">3.5</span> Data Gathered with Small Moving Platforms</a>
  <ul class="collapse">
  <li><a href="#introduction-4" id="toc-introduction-4" class="nav-link" data-scroll-target="#introduction-4"><span class="header-section-number">3.5.1</span> Introduction</a></li>
  <li><a href="#recommendations-for-data-packages-4" id="toc-recommendations-for-data-packages-4" class="nav-link" data-scroll-target="#recommendations-for-data-packages-4"><span class="header-section-number">3.5.2</span> Recommendations for data packages</a></li>
  <li><a href="#resources-3" id="toc-resources-3" class="nav-link" data-scroll-target="#resources-3"><span class="header-section-number">3.5.3</span> Resources</a></li>
  </ul></li>
  <li><a href="#data-in-other-repositories" id="toc-data-in-other-repositories" class="nav-link" data-scroll-target="#data-in-other-repositories"><span class="header-section-number">3.6</span> Data in Other Repositories</a>
  <ul class="collapse">
  <li><a href="#introduction-5" id="toc-introduction-5" class="nav-link" data-scroll-target="#introduction-5"><span class="header-section-number">3.6.1</span> Introduction</a></li>
  <li><a href="#recommendations-for-data-packages-5" id="toc-recommendations-for-data-packages-5" class="nav-link" data-scroll-target="#recommendations-for-data-packages-5"><span class="header-section-number">3.6.2</span> Recommendations for data packages</a></li>
  <li><a href="#nucleotide-sequence-and-genomic-data" id="toc-nucleotide-sequence-and-genomic-data" class="nav-link" data-scroll-target="#nucleotide-sequence-and-genomic-data"><span class="header-section-number">3.6.3</span> Nucleotide sequence and genomic data</a></li>
  <li><a href="#example-data-packages-in-edi-2" id="toc-example-data-packages-in-edi-2" class="nav-link" data-scroll-target="#example-data-packages-in-edi-2"><span class="header-section-number">3.6.4</span> Example data packages in EDI</a></li>
  <li><a href="#appendix-tips-and-repository-information" id="toc-appendix-tips-and-repository-information" class="nav-link" data-scroll-target="#appendix-tips-and-repository-information"><span class="header-section-number">3.6.5</span> Appendix: Tips and repository information</a></li>
  </ul></li>
  <li><a href="#large-data-sets" id="toc-large-data-sets" class="nav-link" data-scroll-target="#large-data-sets"><span class="header-section-number">3.7</span> Large Data Sets</a>
  <ul class="collapse">
  <li><a href="#introduction-6" id="toc-introduction-6" class="nav-link" data-scroll-target="#introduction-6"><span class="header-section-number">3.7.1</span> Introduction</a></li>
  <li><a href="#recommendations-for-data-packages-6" id="toc-recommendations-for-data-packages-6" class="nav-link" data-scroll-target="#recommendations-for-data-packages-6"><span class="header-section-number">3.7.2</span> Recommendations for data packages</a></li>
  <li><a href="#references-1" id="toc-references-1" class="nav-link" data-scroll-target="#references-1"><span class="header-section-number">3.7.3</span> References</a></li>
  <li><a href="#potential-issues" id="toc-potential-issues" class="nav-link" data-scroll-target="#potential-issues"><span class="header-section-number">3.7.4</span> Potential Issues</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/ediorg/dataset_preparation_guides/edit/main/guide-special-cases/dataset-design-special-cases.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ediorg/dataset_preparation_guides/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../guide-special-cases/preface.html">Data Package Design for Special Cases</a></li><li class="breadcrumb-item"><a href="../guide-special-cases/dataset-design-special-cases.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Introduction</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Introduction</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>Members of the working group developing these documents: S. Beaulieu, R. Brown, J. Downing, S. Elmendorf, H. Garritt, G. Gastil-Buhl, C. Gries, J. Hollingsworth, H.-Y. Hsieh, L. Kui, M. Martin, G. Maurer, A. Nguyen, J. Porter, A. Sapp, M. Servilla, T. Whiteaker</p>
<p>In these documents we consider special cases for archiving research data based on their data type, format, or acquisition method, and recommend practices that ensure optimal re-usability of the data. Most recommendations here are aimed at improving documentation of data acquisition and processing to avoid misinterpretation. This includes the recommendation to publish raw data and/or processing code along with the data products. Others are aimed at usability in terms of data size/volume, or connecting related data. Some recommendations involve including a metadata document formatted according to a new and emerging standard (e.g., codeMeta) or a data inventory table. Data inventory tables can cross the line between metadata and data and are intended to improve discoverability and navigation of archived data.</p>
<p>The intended audience for these best practice recommendations is the ecological research information manager (IM) community, and they are applicable to anyone operating in the context of an ecological research program. We assume that the target data repository is designed to handle ecological data, and that a given archive package will include metadata encoded in a community standard. This document references elements of the <a href="https://eml.ecoinformatics.org/">EML metadata standard</a>, but many aspects would similarly apply to other metadata standards and these documents should be considered in the larger context of applicable metadata standard best practices. We refer to the Environmental Data Initiative (<a href="https://portal.edirepository.org">EDI</a>) as an example data repository, though the same practices could be applied to other similar repositories.</p>
<p>Throughout the chapters we use the term <em>data package</em> to refer to a published unit of data and metadata together, which is the convention at the EDI repository. At other data repositories, equivalent terms for a data package, such as <em>dataset</em>, may be used. A data package may contain one or more <em>entities</em>, such as csv tables, spatial data, processing or modeling code, and other documents (pdf, jpg, zip). A basic discussion of data package design can be found as <a href="https://edirepository.org/resources/designing-a-data-package">EDI’s first phase of data publishing documentation</a> and in the <a href="https://environmentaldatainitiative.files.wordpress.com/2017/11/emlbestpractices-v3.pdf">LTER Best Practices for Dataset Metadata in Ecological Metadata Language (EML)</a>.</p>
<p>Generally, we recommend archiving entities using standard file formats that are likely to be machine readable in the future. Exceptions to this may exist where the community standard for processing particular data types relies on specialized file formats (binary, closed specification, etc.) or proprietary software. In these cases, it may be appropriate to archive specialized file types and/or a copy that has been parsed into a format (e.g.&nbsp;ascii) that does not require proprietary software.</p>
<p><strong>Table of contents</strong></p>
<ul>
<li><a href="./code.html">Processing code</a></li>
<li><a href="./model-based-datasets.html">Modeling datasets</a></li>
<li><a href="./images-and-documents-as-data.html">Images and documents</a></li>
<li><a href="./spatial-data.html">Spatial data</a></li>
<li><a href="./data-gathered-with-small-moving-platforms.html">Data gathered with small, moving platforms</a></li>
<li><a href="./data-in-other-repositories.html">Provenance and data in other repositories</a></li>
<li><a href="./large-data-sets.html">Very large datasets</a></li>
</ul>
<section id="code" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="code"><span class="header-section-number">3.1</span> Code</h2>
<p>Contributors: An T. Nguyen, Tim Whiteaker</p>
<section id="introduction" class="level3" data-number="3.1.1">
<h3 data-number="3.1.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">3.1.1</span> Introduction</h3>
<p>This document describes best practices for archiving software, code, or scripts, such as a simulation model, data visualization package, or data manipulation scripts. The intention of these recommendations is to make research based on modeling or software more transparent rather than achieve exact reproducibility, i.e., provide sufficient documentation so that a knowledgeable person can understand algorithms, programming decisions, and their ramifications for the results, rather than run the model and obtain the same results.</p>
<p>Examples of candidate archives for code include <a href="https://www.comses.net/">CoMSES Net</a>, which focuses on sharing models related to social and ecological sciences, and <a href="https://zenodo.org/">Zenodo</a>, a popular DOI-minting all-purpose repository, that can conveniently archive a specific version of <a href="https://guides.github.com/activities/citable-code/">code in a GitHub repository</a>. Alternatively, code may be archived in the EDI repository, either by itself or as part of a data package. The best practices in this document cover both archiving code in EDI and referencing code archived elsewhere.</p>
<p>While metadata for software may be described in detail using the EML &lt;<strong>software</strong>&gt; tree, there exists a project called <a href="https://codemeta.github.io/index.html">CodeMeta</a> which is specifically designed for software metadata. Therefore, one of the key recommendations in this document is to include a CodeMeta file when archiving software or code in EDI.</p>
</section>
<section id="recommendations-for-data-packages" class="level3" data-number="3.1.2">
<h3 data-number="3.1.2" class="anchored" data-anchor-id="recommendations-for-data-packages"><span class="header-section-number">3.1.2</span> Recommendations for data packages</h3>
<section id="considerations-for-archiving-software-or-code" class="level4" data-number="3.1.2.1">
<h4 data-number="3.1.2.1" class="anchored" data-anchor-id="considerations-for-archiving-software-or-code"><span class="header-section-number">3.1.2.1</span> Considerations for archiving software or code</h4>
<ul>
<li>If it is a model and/or a model-based dataset, please see the <a href="model-based-datasets.html">best practices for archiving model-based datasets</a>.</li>
<li>How likely is it that the code will be well maintained into the future? For example, code packages submitted to established code repositories may stay there only while they comply with all testing requirements and may be removed if not well maintained (e.g., the R package repository CRAN). If that commitment to code maintenance is unlikely, such a package should be archived in a repository without maintenance requirements.</li>
<li>Should the code be archived as a separate package or with the data?
<ul>
<li>If the code is used to generate several independent datasets it should be archived as a separate package.</li>
<li>The software authors wishing to place it under a different license from that of the associated data, or to obtain a DOI for only the code, may be reasons to separate code and data packages.</li>
<li>If deciding to package code separately, it may be archived on EDI or another repository. If archiving code outside of EDI, see section 2.2.4 for instructions on how to reference that code from related data packages in EDI.</li>
<li>In most other cases, it is recommended to archive code and data together for context.</li>
</ul></li>
<li>Large community software packages are usually maintained and available elsewhere. However, they may undergo significant updates and it may make sense to archive the code of a certain version with the data for transparency reasons. Consider whether prior versions of a software package are available wherever that software is distributed.</li>
<li>When choosing a repository for the code, consider the ease of the archiving process and how well the code can be described. For example, Zenodo offers an easy pathway to archive code that is currently in GitHub, though metadata requirements are very light. Following the best practices described herein, you would create a CodeMeta file if you were going to archive with EDI. This is more rigorous than Zenodo, but then your code is better described, and in a machine-readable way.</li>
</ul>
</section>
<section id="documenting-softwarecode" class="level4" data-number="3.1.2.2">
<h4 data-number="3.1.2.2" class="anchored" data-anchor-id="documenting-softwarecode"><span class="header-section-number">3.1.2.2</span> Documenting software/code</h4>
<p>When describing the code with EML, include the code as an otherEntity in a data package. Although a well documented human readable text format of the code is preferred, in case of multiple scripts, and/or where directory structure is important, a zip archive may be used. For the formatName and entityType elements in EML, we recommend using format names from the <a href="https://cn.dataone.org/cn/v2/formats">DataONE format list</a> when possible. Some format names are included in examples below. Always check the list for the most up-to-date version of these names.</p>
<p>Example 1: EML otherEntity snippet for a script file.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode xml code-with-copy"><code class="sourceCode xml"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>&lt;<span class="kw">otherEntity</span>&gt;</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">entityName</span>&gt;R script to process CTD data&lt;/<span class="kw">entityName</span>&gt;</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">entityDescription</span>&gt;Annotated RMarkdown script to process, calibrate, and flag raw CTD data.&lt;/<span class="kw">entityDescription</span>&gt;</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">physical</span>&gt;</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">objectName</span>&gt;BLE_LTER_CTD_QAQC.Rmd&lt;/<span class="kw">objectName</span>&gt;</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">size</span><span class="ot"> unit=</span><span class="st">"byte"</span>&gt;9674&lt;/<span class="kw">size</span>&gt;</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">authentication</span><span class="ot"> method=</span><span class="st">"MD5"</span>&gt;8547b7a63fcf6c1f0913a5bd7549d9d1&lt;/<span class="kw">authentication</span>&gt;</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">dataFormat</span>&gt;</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">externallyDefinedFormat</span>&gt;</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>            &lt;<span class="kw">formatName</span>&gt;R Markdown file&lt;/<span class="kw">formatName</span>&gt;</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>         &lt;/<span class="kw">externallyDefinedFormat</span>&gt;</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>      &lt;/<span class="kw">dataFormat</span>&gt;</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>   &lt;/<span class="kw">physical</span>&gt;</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">entityType</span>&gt;script&lt;/<span class="kw">entityType</span>&gt;</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>&lt;/<span class="kw">otherEntity</span>&gt;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="software-license" class="level5" data-number="3.1.2.2.1">
<h5 data-number="3.1.2.2.1" class="anchored" data-anchor-id="software-license"><span class="header-section-number">3.1.2.2.1</span> Software License</h5>
<p>It is important to include a use license to make it clear how others can use your work. We recommend the <a href="https://creativecommons.org/share-your-work/public-domain/cc0/">Creative Commons “no copyright reserved” (CC0)</a> license, which places the software in the public domain and makes it easiest for end users to adapt and use your work. If a more restrictive license is required, we recommend the <a href="https://directory.fsf.org/wiki/License:Apache-2.0">Apache License, Version 2.0</a> license, a permissive license that allows others to reuse, modify, and redistribute your software.</p>
<p>If a mix of data and code needs to be archived, and they each fall under different licenses, then separating them into different packages is advisable to eliminate ambiguity on which license applies to which portion of a data package. When a license other than a public domain dedication is used, then in addition to specifying the license in the metadata (see the “intellectualRights” element in EML), consider including a copy of the license at the beginning of the code files themselves so that the license is readily apparent to end users who peruse the code.</p>
</section>
<section id="codemeta" class="level5" data-number="3.1.2.2.2">
<h5 data-number="3.1.2.2.2" class="anchored" data-anchor-id="codemeta"><span class="header-section-number">3.1.2.2.2</span> CodeMeta</h5>
<p>Include a CodeMeta JSON file for all code that is archived in EDI. The CodeMeta file should be named “codemeta.json” and listed as an EML otherEntity. The formatName should be “JavaScript Object Notation (JSON) file”, the entityType should be “metadata”, and the entityDescription should indicate that this is a CodeMeta file for a given software or script in the data package.</p>
<p>For unnamed projects, e.g., one-off scripts for data processing, analysis, and/or visualisation, a CodeMeta file might appear to be overkill; however, CodeMeta files are simple to generate, and we recommend the below bare minimum. If there are multiple scripts each in their own otherEntity tag, we recommend aggregating information about them into one codemeta.json.</p>
<p>Example 2: Minimum recommended codemeta.json example for unnamed projects.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">{</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"@context"</span><span class="fu">:</span> <span class="ot">[</span><span class="st">"https://doi.org/10.5063/schema/codemeta-2.0"</span><span class="ot">,</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>      <span class="st">"http://schema.org"</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>   <span class="ot">]</span><span class="fu">,</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"@type"</span><span class="fu">:</span> <span class="st">"SoftwareSourceCode"</span><span class="fu">,</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"description"</span><span class="fu">:</span> <span class="st">"RMarkdown script to calibrate and flag raw CTD data."</span><span class="fu">,</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"author"</span><span class="fu">:</span> <span class="fu">{</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"@type"</span><span class="fu">:</span> <span class="st">"Person"</span><span class="fu">,</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"givenName"</span><span class="fu">:</span> <span class="st">"Christina"</span><span class="fu">,</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"familyName"</span><span class="fu">:</span> <span class="st">"Bonsell"</span><span class="fu">,</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"email"</span><span class="fu">:</span> <span class="st">"cbonsell@utexas.edu"</span><span class="fu">,</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"@id"</span><span class="fu">:</span> <span class="st">"https://orcid.org/0000-0002-8564-0618"</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>   <span class="fu">},</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"keywords"</span><span class="fu">:</span> <span class="ot">[</span><span class="st">"calibration"</span><span class="ot">,</span> <span class="st">"CTD"</span><span class="ot">,</span> <span class="st">"RMarkdown"</span><span class="ot">]</span><span class="fu">,</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"license"</span><span class="fu">:</span> <span class="st">"https://unlicense.org/"</span><span class="fu">,</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"dateCreated"</span><span class="fu">:</span> <span class="st">"2013-10-19"</span><span class="fu">,</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"programmingLanguage"</span><span class="fu">:</span> <span class="fu">{</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"@type"</span><span class="fu">:</span> <span class="st">"ComputerLanguage"</span><span class="fu">,</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"name"</span><span class="fu">:</span> <span class="st">"R"</span><span class="fu">,</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"version"</span><span class="fu">:</span> <span class="st">"3.6.2"</span><span class="fu">,</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"url"</span><span class="fu">:</span> <span class="st">"https://r-project.org"</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>   <span class="fu">}</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="fu">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Example 3: sample otherEntity metadata for example 2â€™s codemeta.json.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode xml code-with-copy"><code class="sourceCode xml"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>&lt;<span class="kw">otherEntity</span>&gt;</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">entityName</span>&gt;CodeMeta file for BLE_LTER_CTD_QAQC.Rmd&lt;/<span class="kw">entityName</span>&gt;</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">entityDescription</span>&gt;CodeMeta file for annotated RMarkdown script to process, calibrate, and flag raw CTD data.&lt;/<span class="kw">entityDescription</span>&gt;</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">physical</span>&gt;</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">objectName</span>&gt;codemeta.json&lt;/<span class="kw">objectName</span>&gt;</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">size</span><span class="ot"> unit=</span><span class="st">"byte"</span>&gt;702&lt;/<span class="kw">size</span>&gt;</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">authentication</span><span class="ot"> method=</span><span class="st">"MD5"</span>&gt;8547b7a63abc6c1f0913a5bd7549d9d1&lt;/<span class="kw">authentication</span>&gt;</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">dataFormat</span>&gt;</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">externallyDefinedFormat</span>&gt;</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>            &lt;<span class="kw">formatName</span>&gt;application/json&lt;/<span class="kw">formatName</span>&gt;</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>         &lt;/<span class="kw">externallyDefinedFormat</span>&gt;</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>      &lt;/<span class="kw">dataFormat</span>&gt;</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>   &lt;/<span class="kw">physical</span>&gt;</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">entityType</span>&gt;CodeMeta&lt;/<span class="kw">entityType</span>&gt;</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>&lt;/<span class="kw">otherEntity</span>&gt;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>For named projects, also include the software name, and the version if applicable. The example below shows some additional metadata you can include. See also the more complete <a href="https://docs.ropensci.org/codemetar/articles/codemeta-intro.html">codemetar example</a> and the available <a href="https://codemeta.github.io/terms/">CodeMeta terms</a>.</p>
<p>Example 4: A more complete CodeMeta example for named projects. Example taken from the CodeMeta project Github with edits for brevity.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">{</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"@context"</span><span class="fu">:</span> <span class="ot">[</span><span class="st">"https://doi.org/10.5063/schema/codemeta-2.0"</span><span class="ot">,</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>      <span class="st">"http://schema.org"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>   <span class="ot">]</span><span class="fu">,</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"@type"</span><span class="fu">:</span> <span class="st">"SoftwareSourceCode"</span><span class="fu">,</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"name"</span><span class="fu">:</span> <span class="st">"codemetar: Generate 'CodeMeta' Metadata for R Packages"</span><span class="fu">,</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"description"</span><span class="fu">:</span> <span class="st">"A JSON-LD format for software metadata"</span><span class="fu">,</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"author"</span><span class="fu">:</span> <span class="ot">[</span><span class="fu">{</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"@type"</span><span class="fu">:</span> <span class="st">"Person"</span><span class="fu">,</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"givenName"</span><span class="fu">:</span> <span class="st">"Carl"</span><span class="fu">,</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"familyName"</span><span class="fu">:</span> <span class="st">"Boettiger"</span><span class="fu">,</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"email"</span><span class="fu">:</span> <span class="st">"cboettig@gmail.com"</span><span class="fu">,</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"@id"</span><span class="fu">:</span> <span class="st">"https://orcid.org/0000-0002-1642-628X"</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>      <span class="fu">}</span><span class="ot">,</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>      <span class="fu">{</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"@type"</span><span class="fu">:</span> <span class="st">"Person"</span><span class="fu">,</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"givenName"</span><span class="fu">:</span> <span class="st">"MaÃ«lle"</span><span class="fu">,</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"familyName"</span><span class="fu">:</span> <span class="st">"Salmon"</span><span class="fu">,</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"@id"</span><span class="fu">:</span> <span class="st">"https://orcid.org/0000-0002-2815-0399"</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">}</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>   <span class="ot">]</span><span class="fu">,</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"codeRepository"</span><span class="fu">:</span> <span class="st">"https://github.com/ropensci/codemetar"</span><span class="fu">,</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"dateCreated"</span><span class="fu">:</span> <span class="st">"2013-10-19"</span><span class="fu">,</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"license"</span><span class="fu">:</span> <span class="st">"https://spdx.org/licenses/GPL-3.0"</span><span class="fu">,</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"version"</span><span class="fu">:</span> <span class="st">"0.1.8"</span><span class="fu">,</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"programmingLanguage"</span><span class="fu">:</span> <span class="fu">{</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"@type"</span><span class="fu">:</span> <span class="st">"ComputerLanguage"</span><span class="fu">,</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"name"</span><span class="fu">:</span> <span class="st">"R"</span><span class="fu">,</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"version"</span><span class="fu">:</span> <span class="st">"3.5.3"</span><span class="fu">,</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>      <span class="dt">"url"</span><span class="fu">:</span> <span class="st">"https://r-project.org"</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>   <span class="fu">},</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"softwareRequirements"</span><span class="fu">:</span> <span class="ot">[</span><span class="fu">{</span></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"@type"</span><span class="fu">:</span> <span class="st">"SoftwareApplication"</span><span class="fu">,</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"identifier"</span><span class="fu">:</span> <span class="st">"R"</span><span class="fu">,</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"name"</span><span class="fu">:</span> <span class="st">"R"</span><span class="fu">,</span></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"version"</span><span class="fu">:</span> <span class="st">"&gt;= 3.0.0"</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>      <span class="fu">}</span><span class="ot">,</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>      <span class="fu">{</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"@type"</span><span class="fu">:</span> <span class="st">"SoftwareApplication"</span><span class="fu">,</span></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"identifier"</span><span class="fu">:</span> <span class="st">"git2r"</span><span class="fu">,</span></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"name"</span><span class="fu">:</span> <span class="st">"git2r"</span><span class="fu">,</span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>         <span class="dt">"provider"</span><span class="fu">:</span> <span class="fu">{</span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>            <span class="dt">"@id"</span><span class="fu">:</span> <span class="st">"https://cran.r-project.org"</span><span class="fu">,</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>            <span class="dt">"@type"</span><span class="fu">:</span> <span class="st">"Organization"</span><span class="fu">,</span></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>            <span class="dt">"name"</span><span class="fu">:</span> <span class="st">"Comprehensive R Archive Network (CRAN)"</span><span class="fu">,</span></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>            <span class="dt">"url"</span><span class="fu">:</span> <span class="st">"https://cran.r-project.org"</span></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>         <span class="fu">}</span></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>      <span class="fu">}</span></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>   <span class="ot">]</span><span class="fu">,</span></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>   <span class="dt">"keywords"</span><span class="fu">:</span> <span class="ot">[</span><span class="st">"metadata"</span><span class="ot">,</span> <span class="st">"codemeta"</span><span class="ot">,</span> <span class="st">"ropensci"</span><span class="ot">]</span></span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a><span class="fu">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="metadata-to-enable-reproducibility" class="level5" data-number="3.1.2.2.3">
<h5 data-number="3.1.2.2.3" class="anchored" data-anchor-id="metadata-to-enable-reproducibility"><span class="header-section-number">3.1.2.2.3</span> Metadata to enable reproducibility</h5>
<p>When archiving software, we strongly recommend including a user guide with installation and usage instructions if such would not already be apparent to the typical user. Take into account that the user might not have access to certain inputs that the software/scripts require. Include when feasible at least some example data, and configure the script so that it is ready to run with the example data.</p>
<p>Aside from the software/code itself and its dependencies, other pieces of information may be important should a user wish to reproduce results, such as the operating system and version and the system locale. Include this information in the data packageâ€™s methods/methodStep/description. For certain tools, there are ways to easily generate this information, e.g., a call to sessionInfo() in the R console. If the system outputs this information in a standardly formatted plain text file, that might be included as an otherEntity.</p>
</section>
</section>
<section id="linking-code-and-data" class="level4" data-number="3.1.2.3">
<h4 data-number="3.1.2.3" class="anchored" data-anchor-id="linking-code-and-data"><span class="header-section-number">3.1.2.3</span> Linking code and data</h4>
<p>There are a few solutions for providing explicit machine-readable linkages between different entities/packages (the distinction between code/data doesnâ€™t matter too much here). For most cases we recommend the simplest approach, which is to use the methods/methodStep/description element of EML. More advanced users may wish to utilize the other solutions described herein.</p>
<section id="descriptive-approach" class="level5" data-number="3.1.2.3.1">
<h5 data-number="3.1.2.3.1" class="anchored" data-anchor-id="descriptive-approach"><span class="header-section-number">3.1.2.3.1</span> Descriptive approach</h5>
<p>In the dataset methods/methodStep/description element, include verbal descriptions such as “results.csv was derived from raw_data.csv using script.R” and repeat for all entities. If code and data reside in different packages, be sure to specify that.</p>
</section>
<section id="the-eml-datasource-element" class="level5" data-number="3.1.2.3.2">
<h5 data-number="3.1.2.3.2" class="anchored" data-anchor-id="the-eml-datasource-element"><span class="header-section-number">3.1.2.3.2</span> The EML dataSource element</h5>
<p>Nested under methods/methodStep, dataSource elements describe other data packages that serve as source for the current package. dataSource looks like a mini-EML tree describing the source data. Example: <a href="http://portal.edirepository.org/nis/simpleSearch?defType=edismax&amp;q=ecocomDP&amp;fq=-scope:ecotrends&amp;fq=-scope:lter-landsat*&amp;fl=id,packageid,title,author,organization,pubdate,coordinates&amp;debug=false">ecocomDP packages</a> list the original packages under dataSource. dataSource does not describe relationships between entities in the same package, and as far as we know there is no explicit way in EML to do so.</p>
</section>
<section id="provone" class="level5" data-number="3.1.2.3.3">
<h5 data-number="3.1.2.3.3" class="anchored" data-anchor-id="provone"><span class="header-section-number">3.1.2.3.3</span> ProvONE</h5>
<p><a href="http://jenkins-1.dataone.org/jenkins/view/Documentation%20Projects/job/ProvONE-Documentation-trunk/ws/provenance/ProvONE/v1/provone.html">ProvONE</a> is a model developed by DataONE affiliates for provenance or denoting relationships between data entities. Each package on DataONE is described by a science metadata document (e.g., EML, ISO, FGDC) and a resource map document following ProvONE. The resource map powers a nice display of data relationships (see <a href="https://arcticdata.io/catalog/view/doi%3A10.18739%2FA2556Q">this package on the Arctic Data Center</a>). This handles both relationships between entities in the same package and entities residing in different packages. However, note that EDI currently does not utilize this model.</p>
</section>
</section>
<section id="external-software" class="level4" data-number="3.1.2.4">
<h4 data-number="3.1.2.4" class="anchored" data-anchor-id="external-software"><span class="header-section-number">3.1.2.4</span> External software</h4>
<p>Large community-backed tools or proprietary software such as ArcGIS Pro or Microsoft Excel do not need to be archived. However, if they have had any impact on the final data (e.g., ArcGIS Pro was used to modify spatial rasters), the EML methods section should describe the routines performed. Within the data package, indicate linkage to external software as follows.</p>
<ul>
<li>Briefly describe the software/code and its relationship to the data in EMLâ€™s methods/methodStep/description element.</li>
<li>Names of all software used. Include both the common acronym and the full spelling.</li>
<li>The URL(s) to all models/software used. Stable, persistent URLs pointing to exact version(s) are preferable, rather than generic links such as a project homepage. If the archived model has a DOI, then include a full citation to the model in the methods/methodStep/description text. The exception to this is when referencing tools such as Excel that have achieved global household name status.</li>
<li>Broadly, the system setup used, if relevant.</li>
<li>Information on exact versions for all code used (including dependencies). This is important, e.g., ArcGIS Pro 2.4.1 is very different from ArcGIS for Desktop 10.7.1. Different systems have methods to easily generate this information, e.g.&nbsp;a call to sessionInfo() in the R console.</li>
<li>Consider, if applicable, to archive the “runfile” as its own data entity within the data package, i.e., the script(s) that sets parameters and/or calls on functions imported from external software.</li>
</ul>
<p>Example 5: EML method description referring to external software.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode xml code-with-copy"><code class="sourceCode xml"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>&lt;<span class="kw">methods</span>&gt;</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">methodStep</span>&gt;</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">description</span>&gt;</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">para</span>&gt;           </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>            The seagrass coverage raster was created in ArcGIS Pro (version</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>            2.4.3, by Esri) using the IDW geoprocessing tool on</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>            sampling_points.csv with a power of 2 and the nearest</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>            12 points.</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>         &lt;/<span class="kw">para</span>&gt;</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">para</span>&gt;</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>            The raster was then refined using the seagrass-refiner package</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>            with the auto-refine option checked (Smith, 2017).</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>         &lt;/<span class="kw">para</span>&gt;</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">para</span>&gt;</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>            Smith, J. (2017). seagrass-refiner: a package that does the cool</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>            seagrass stuff, Version 1.2, Zenodo.  https://doi.org/this-is/a-fake-doi,</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>            2017.</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>         &lt;/<span class="kw">para</span>&gt;</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>      &lt;/<span class="kw">description</span>&gt;</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>   &lt;/<span class="kw">methodStep</span>&gt;</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>&lt;/<span class="kw">methods</span>&gt;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="resources" class="level3" data-number="3.1.3">
<h3 data-number="3.1.3" class="anchored" data-anchor-id="resources"><span class="header-section-number">3.1.3</span> Resources</h3>
<p><a href="https://codemeta.github.io/">CodeMeta</a> website</p>
<p><a href="https://codemeta.github.io/codemeta-generator/">CodeMeta generator</a> for creating CodeMeta</p>
<p><a href="https://codemeta.github.io/crosswalk/">CodeMeta crosswalks</a> for a number of popular software</p>
<p><a href="https://codemeta.github.io/terms/">CodeMeta terms</a> you can use for describing software</p>
<p>A description of some <a href="https://opensource.org/licenses">software licenses</a></p>
<p><a href="model-based-datasets.html">Best practices document to archiving model-based datasets</a></p>
<p><a href="http://jenkins-1.dataone.org/jenkins/view/Documentation%20Projects/job/ProvONE-Documentation-trunk/ws/provenance/ProvONE/v1/provone.html">ProvONE documentation</a></p>
<p><a href="https://www.w3.org/TR/prov-o/">W3C PROV-O documentation</a></p>
<p><a href="https://docs.google.com/document/d/1JeznivTDubi0ZX_lsO50eCUl-8zxSiz_xq5SsBRwbuw/edit">Licensing software as part of an EDI data package</a></p>
</section>
</section>
<section id="model-based-datasets" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="model-based-datasets"><span class="header-section-number">3.2</span> Model-Based Datasets</h2>
<p>Contributors: An T. Nguyen, Tim Whiteaker, Corinna Gries</p>
<section id="introduction-1" class="level3" data-number="3.2.1">
<h3 data-number="3.2.1" class="anchored" data-anchor-id="introduction-1"><span class="header-section-number">3.2.1</span> Introduction</h3>
<p>This document includes recommendations for archiving data packages composed of model-based datasets. These datasets may include the model code itself, input data, model parameter settings, and output data.</p>
<p>The range of cases for model-based datasets includes small one-off model code specific to one research question, through various code packages which are maintained in community repositories as long as they meet requirements (e.g., CRAN for R packages), to large community models maintained by groups of programmers and users.</p>
<p>The intention of these recommendations is to make research based on modeling more transparent rather than achieve exact reproducibility, i.e., provide sufficient documentation so that a knowledgeable person can understand algorithms, programming decisions, and their ramifications for the results, rather than run the model and obtain the same results.</p>
<p>It is not always easy to determine who among project personnel (IMs, scientists, programmers) is responsible for the different components of a model-based dataset. This is best decided on a case-by-case basis. A common division is that the code authors annotate the code, and the IM handles the archiving and linkage to data product(s); partially except in cases of large community models.</p>
</section>
<section id="recommendations-for-data-packages-1" class="level3" data-number="3.2.2">
<h3 data-number="3.2.2" class="anchored" data-anchor-id="recommendations-for-data-packages-1"><span class="header-section-number">3.2.2</span> Recommendations for data packages</h3>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/model_datasets.png" title="model data management flowchart" class="img-fluid figure-img"></p>
<figcaption>Figure 1: Flowchart for considering archival paths for various model components.</figcaption>
</figure>
</div>
<section id="referencing-models-in-eml" class="level4" data-number="3.2.2.1">
<h4 data-number="3.2.2.1" class="anchored" data-anchor-id="referencing-models-in-eml"><span class="header-section-number">3.2.2.1</span> Referencing models in EML</h4>
<p>For data packages related to a model, whether the model is archived within the same data package or not, indicate linkage to the model in EML following the <a href="code.html">best practices for archiving code</a> (see the section on linking code and data).</p>
<p>Example 1: EML snippet relating data to models via the method description:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode xml code-with-copy"><code class="sourceCode xml"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>&lt;<span class="kw">methodStep</span>&gt;</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">description</span>&gt;</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">para</span>&gt;This methodStep contains data provenance information as specified in the LTER EML Best Practices. Each dataSource element here lists entity-specific information and links to source data used in the creation of this derivative data package.&lt;/<span class="kw">para</span>&gt;</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>   &lt;/<span class="kw">description</span>&gt;</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">dataSource</span>&gt;</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">title</span>&gt;Source dataset title&lt;/<span class="kw">title</span>&gt;</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">creator</span>&gt;</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">individualName</span>&gt;</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>            &lt;<span class="kw">givenName</span>&gt;first name&lt;/<span class="kw">givenName</span>&gt;</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>            &lt;<span class="kw">surName</span>&gt;last name&lt;/<span class="kw">surName</span>&gt;</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>         &lt;/<span class="kw">individualName</span>&gt;</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">organizationName</span>&gt;organization name&lt;/<span class="kw">organizationName</span>&gt;</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">electronicMailAddress</span>&gt;email@some.edu&lt;/<span class="kw">electronicMailAddress</span>&gt;</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>      &lt;/<span class="kw">creator</span>&gt;</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">distribution</span>&gt;</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">online</span>&gt;</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>            &lt;<span class="kw">onlineDescription</span>&gt;This is a link to an external online data resource (describe resource and repository location).&lt;/<span class="kw">onlineDescription</span>&gt;</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>            &lt;<span class="kw">url</span><span class="ot"> function=</span><span class="st">"information"</span>&gt;https://pasta.lternet.edu/package/metadata/eml/knb-lter-ntl/80/2&lt;/<span class="kw">url</span>&gt;</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>         &lt;/<span class="kw">online</span>&gt;</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>      &lt;/<span class="kw">distribution</span>&gt;</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">contact</span>&gt;</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">positionName</span>&gt;Information Manager&lt;/<span class="kw">positionName</span>&gt;</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">organizationName</span>&gt;organization name&lt;/<span class="kw">organizationName</span>&gt;</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">electronicMailAddress</span>&gt;infomgr@some.edu&lt;/<span class="kw">electronicMailAddress</span>&gt;</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>      &lt;/<span class="kw">contact</span>&gt;</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>   &lt;/<span class="kw">dataSource</span>&gt;</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>&lt;/<span class="kw">methodStep</span>&gt;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="model-code" class="level5" data-number="3.2.2.1.1">
<h5 data-number="3.2.2.1.1" class="anchored" data-anchor-id="model-code"><span class="header-section-number">3.2.2.1.1</span> Model code</h5>
<p>The model used to produce certain data needs to be well documented and linked from the resulting data product(s). However, it is not always easy to decide where and how to archive the code, and whether or not in conjunction with the data product(s). We outline in sections below three common code archiving options.</p>
<p>Note that these scenarios (model code archived with data, or standalone in EDI, or elsewhere) are not mutually exclusive. Any project that involves code might make use of both established and custom software hosted on many different platforms, and might use some or all archiving options.</p>
<p>To decide between archiving options, consider the questions listed in <a href="code.html">best practices for publishing code</a>.</p>
</section>
<section id="model-code-and-data-in-the-same-package" class="level5" data-number="3.2.2.1.2">
<h5 data-number="3.2.2.1.2" class="anchored" data-anchor-id="model-code-and-data-in-the-same-package"><span class="header-section-number">3.2.2.1.2</span> Model code and data in the same package</h5>
<p>The goal of this practice is to ensure transparency of the data, and it applies to one-off models developed for the associated data, or occasionally to larger code bases for the reasons outlined in <a href="code.html">best practices for archiving code</a>. Include the code as a dataset/otherEntity. Additionally, it is recommended to include a CodeMeta file, which can also be handled and documented in EML as dataset/otherEntity. CodeMeta is a metadata standard for software and code compatible with schema.org. Refer to <a href="code.html">best practices for archiving code</a> for how to document the code and create CodeMeta.</p>
</section>
<section id="model-code-as-standalone-package" class="level5" data-number="3.2.2.1.3">
<h5 data-number="3.2.2.1.3" class="anchored" data-anchor-id="model-code-as-standalone-package"><span class="header-section-number">3.2.2.1.3</span> Model code as standalone package</h5>
<p>If the model has been used to generate several datasets, i.e., is more widely applicable, it can be archived as its own package in EDI and assigned a DOI. Include the code as a dataset/otherEntity. Additionally, it is recommended to include a CodeMeta JSON-LD file, which can also be handled and documented in EML as dataset/otherEntity. CodeMeta is a metadata standard for software and code compatible with schema.org. Refer to <a href="code.html">best practices for archiving code</a> for how to document the code and create CodeMeta.</p>
</section>
<section id="model-code-archivedmaintained-elsewhere" class="level5" data-number="3.2.2.1.4">
<h5 data-number="3.2.2.1.4" class="anchored" data-anchor-id="model-code-archivedmaintained-elsewhere"><span class="header-section-number">3.2.2.1.4</span> Model code archived/maintained elsewhere</h5>
<p>This might include complex community models/software maintained by many people, published and actively maintained R/Python packages, etc., or simply code archived in another repository such as <a href="https://www.comses.net/">CoMSES Net</a>. It may sometimes be advisable to archive a copy of the model code with the data, even if it appears to be maintained elsewhere. See recommendations above for referencing models in EML.</p>
</section>
</section>
<section id="model-input-and-output-data" class="level4" data-number="3.2.2.2">
<h4 data-number="3.2.2.2" class="anchored" data-anchor-id="model-input-and-output-data"><span class="header-section-number">3.2.2.2</span> Model input and output data</h4>
<p>These are considered data entities, which should be handled according to EML best practices for corresponding data types. However, if the resulting datasets are very large, one may consider if input/output from all individual model runs need to be archived. Are there specific model run results that are more useful for non-modelers? For example: results from model runs leading to a journal publication.</p>
<p>Very large model inputs/outputs may need to be archived offline. Refer to <a href="large-data-sets.html">best practices for offline data</a>.</p>
<p>If the model requires a specific folder structure, you can zip model input files within the package to preserve that folder structure. A disadvantage of this approach is that you cannot elegantly describe each file with EML.</p>
<p>The <a href="https://modeldatarcn.github.io/#project-description">EarthCube Research Coordination Network, “What About Model Data?” group</a> is working on a rubric to help you determine how much model output data to save, based on assorted criteria on reproducibility/value of the data. Learn more about that group and their rubric on their <a href="https://modeldatarcn.github.io/">Model Data RCN website</a>.</p>
<p>Researchers at the Department of Energyâ€™s Environmental Systems Science are also working on assessing model archiving needs. In <a href="https://eartharxiv.org/repository/view/260/">this preprint</a>, Simmonds et al.&nbsp;2020 discuss feedback from communications with modellers and propose preliminary solutions. With regards to input/output data, their feedback indicates two opposite opinions: some feel the whole gamut of raw to aggregated outputs needs to be archived, while others advocate for only high-level outputs corresponding to publication figures. They also found that spin-up simulations were not considered a high priority for archiving. See section 2.3 What is worth archiving and for how long does it remain useful?</p>
</section>
<section id="model-parameters" class="level4" data-number="3.2.2.3">
<h4 data-number="3.2.2.3" class="anchored" data-anchor-id="model-parameters"><span class="header-section-number">3.2.2.3</span> Model parameters</h4>
<p>Include model parameters whenever applicable. If code/input/output from multiple model runs are archived, make sure to archive all corresponding sets of parameters, and be explicit in linking the different components together.</p>
<p>Consider archiving model parameter files as their own data object(s) in both their native format and as a text (non-binary) version. If the ‘runfile’ will be archived, consider including the parameters within that file with appropriate annotations.</p>
</section>
</section>
<section id="example-data-packages-in-edi" class="level3" data-number="3.2.3">
<h3 data-number="3.2.3" class="anchored" data-anchor-id="example-data-packages-in-edi"><span class="header-section-number">3.2.3</span> Example data packages in EDI</h3>
<table>
<tbody><tr>
<td>
<strong>Dataset Title</strong>
</td>
<td>
<strong>Description</strong>
</td>
<td>
<strong>EDI Package ID</strong>
</td>
</tr>
<tr>
<td>
<em>North Temperate Lakes LTER General Lake Model Parameter Set for Lake Mendota, Summer 2016 Calibration</em>
</td>
<td>
Parameters for specific GLM runs. GLM is a large community model, not managed and archived in EDI
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?packageid=knb-lter-ntl.348.2">knb-lter-ntl.348.2</a>
</td>
</tr>
<tr>
<td>
<em>SBC LTER: Regional Oceanic Modeling System (ROMS) Setup Files, Code, and Lagrangian Model Setup Files</em>
</td>
<td>
All the necessary code, grid, forcing, initial, and boundary condition files for running the UCLA version of the Regional Oceanic Modeling System (ROMS) for the Santa Barbara Channel
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?scope=knb-lter-sbc&amp;identifier=126">knb-lter-sbc.126.1</a>
</td>
</tr>
<tr>
<td>
<em>Lake thermal structure drives inter-annual variability in summer anoxia dynamics in a eutrophic lake over 37 years</em>
</td>
<td>
Dataset to run a 37-year simulation (1979-2015) of the Lake Mendota lake ecosystem using the vertical 1D GLM-AED2 model.
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?scope=knb-lter-ntl&amp;identifier=396">knb-lter-ntl.396.1</a>
</td>
</tr>
</tbody></table>
</section>
<section id="resources-1" class="level3" data-number="3.2.4">
<h3 data-number="3.2.4" class="anchored" data-anchor-id="resources-1"><span class="header-section-number">3.2.4</span> Resources</h3>
<p>Janssen, Marco A., Lilian Na’ia Alessa, Michael Barton, Sean Bergin and Allen Lee (2008). â€˜Towards a Community Framework for Agent-Based Modellingâ€™. Journal of Artificial Societies and Social Simulation 11(2)6 <a href="http://jasss.soc.surrey.ac.uk/11/2/6.html">http://jasss.soc.surrey.ac.uk/11/2/6.html</a>.</p>
<p>Simmonds, Maegen, William J. Riley, Shreyas Cholia, and Charuleka Varadharajan (2020). ‘Addressing Model Data Archiving Needs for the Department of Energyâ€™s Environmental Systems Science Community’. EarthArXiv (preprint). <a href="https://doi.org/10.31223/osf.io/acdk4">https://doi.org/10.31223/osf.io/acdk4</a>.</p>
<p>See sections 2.3 What is worth archiving and for how long does it remain useful?, discussed above, plus 2.4 Model data archiving protocol, where the authors argue for better standardized reporting format for model data, e.g., top-level metadata and directory structure at a minimum. Section 4.1 Developing Model Data Archiving Guidelines proposes an organization scheme for model data.</p>
</section>
</section>
<section id="images-and-documents-as-data" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="images-and-documents-as-data"><span class="header-section-number">3.3</span> Images and Documents as Data</h2>
<p>Contributors: Renee F. Brown (lead), Stace Beaulieu, Sarah Elmendorf, Gastil Gastil-Buhl, Corinna Gries, Li Kui, Mary Martin, Greg Maurer, John Porter, Tim Whiteaker</p>
<section id="introduction-2" class="level3" data-number="3.3.1">
<h3 data-number="3.3.1" class="anchored" data-anchor-id="introduction-2"><span class="header-section-number">3.3.1</span> Introduction</h3>
<p>This chapter describes best practices for archiving images and other documents as data. The <a href="http://purl.obolibrary.org/obo/IAO_0000101">Environment Ontology (ENVO)</a> defines a document as ‘<em>a collection of information content entities intended to be understood together as a whole</em>.’ Common examples include still images, audio and/or video multimedia files, field notebooks, written interview notes or transcribed oral accounts, historical document collections, and ‘paper’ maps (non-digitized maps). For images that are already handled by specialized repositories (e.g., phenocam images, specimen images) refer to <a href="data-in-other-repositories.html">Data in Other Repositories</a>, for additional information on how to handle images from uncrewed (underwater or aerial) vehicles refer to <a href="data-gathered-with-small-moving-platforms.html">Data Gathered with Small Moving Platforms</a>, and for geospatial imagery refer to <a href="spatial-data.html">Spatial Data</a>.</p>
</section>
<section id="recommendations-for-data-packages-2" class="level3" data-number="3.3.2">
<h3 data-number="3.3.2" class="anchored" data-anchor-id="recommendations-for-data-packages-2"><span class="header-section-number">3.3.2</span> Recommendations for data packages</h3>
<section id="reasons-to-archive-documents-as-data" class="level4" data-number="3.3.2.1">
<h4 data-number="3.3.2.1" class="anchored" data-anchor-id="reasons-to-archive-documents-as-data"><span class="header-section-number">3.3.2.1</span> Reasons to archive documents as data</h4>
<ul>
<li><strong>Enhance the credibility of associated datasets.</strong> Many document types (field notes, still images, etc.) often provide additional metadata that cannot easily be encapsulated in the associated dataset(s) or were not considered important at the time of transcription. As such, these documents may provide opportunities to rectify transcription errors, retrospectively provide explanations of unusual data, and/or include additional observational or measured data, such as opportunistic measurements or calibration parameters.</li>
<li><strong>Provide opportunities for new analyses.</strong> New analytical methods may be employed on archived documents (especially still images) or documents that were never archived previously because the cost-to-benefit ratio was considered too high (e.g., pilot projects).</li>
<li><strong>Improve ease of access.</strong> In distributed projects, access to original and/or ‘hard-copy’ documents may be limited to a particular institution or subset of people. By digitally archiving these documents in a data repository, the data become more findable, accessible, interoperable, and reusable (FAIR).</li>
</ul>
</section>
<section id="considerations-for-data-package-structure" class="level4" data-number="3.3.2.2">
<h4 data-number="3.3.2.2" class="anchored" data-anchor-id="considerations-for-data-package-structure"><span class="header-section-number">3.3.2.2</span> Considerations for data package structure</h4>
<ul>
<li><strong>Balance file size and number of files.</strong> A data package may contain document files individually or bundled as a compressed archive (e.g., zip). The decision of how to bundle documents into compressed archives and then into data packages should be guided by the overall goal of making data usable for the intended purpose of the documents. In most cases, this would involve finding specific documents by, for example, the date or location of the acquisition, or some other aspect of interest. In addition, the effort of documenting documents (each individually vs.&nbsp;in groups) has to be taken into account. Also see <a href="large-data-sets.html">Large Data Sets</a>.</li>
<li><strong>Document grouping.</strong> Data packages, or compressed archives within data packages, may be grouped spatially (e.g., by location) and/or temporally (e.g., by date, season, or year). For example, data outputs from a stationary camera may be archived in annual data packages, each containing monthly compressed archives if the number of images is large. While moving camera outputs may also be archived annually, these data packages may instead include compressed archives containing all still images for a single location.</li>
<li><strong>Document naming.</strong> To maximize searchability, document names should be unique and meaningful for a data reuser. It is recommended that individual documents be named according to their content, and compressed archives include date, location, and other relevant information in the filename.</li>
<li><strong>Data inventory table.</strong> An inventory table providing the structure and organization of the included document entities or groups of documents (see Table 4.1) is recommended, especially for larger collections of documents within a data package. The inventory table serves as an additional source of metadata and may also be used to link specific documents to additional information.</li>
<li><strong>Archival frequency.</strong> One should strive for archiving a fully processed group of documents when no more updates are expected (e.g., after a field season or annually) due to the large volume of documents to be handled repeatedly for each update.</li>
<li><strong>Linking to related data packages.</strong> In the case where the documents are useful to understanding another data package and vice versa (e.g., met station visitation logs and met station time series data), it is recommended to link the complementary data package in the methods section of both datasets. Alternatively, include the document(s) or compressed archive(s) in the existing dataset as otherEntity, as described in the next section.</li>
</ul>
</section>
<section id="documenting-data-packages" class="level4" data-number="3.3.2.3">
<h4 data-number="3.3.2.3" class="anchored" data-anchor-id="documenting-data-packages"><span class="header-section-number">3.3.2.3</span> Documenting data packages</h4>
<section id="ecological-metadata-language" class="level5" data-number="3.3.2.3.1">
<h5 data-number="3.3.2.3.1" class="anchored" data-anchor-id="ecological-metadata-language"><span class="header-section-number">3.3.2.3.1</span> Ecological Metadata Language</h5>
<p>All data packages require good discovery-level metadata in Ecological Metadata Language (EML), which should be assembled using standard documented best practices. Documents (including compressed archives) should be included as otherEntity in the data package (e.g., see Example 4.1). Refer to the most recent version of EML Best Practices (<a href="https://environmentaldatainitiative.files.wordpress.com/2017/11/emlbestpractices-v3.pdf">currently v3</a>) for guidance regarding the formatName and entityType EML elements. If a format for your document type is not covered, it is recommended to use the appropriate <a href="https://github.com/DataONEorg/object-formats">MIME type</a>, if available.</p>
<p>Example 1: EML otherEntity snippet for a pdf file</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode xml code-with-copy"><code class="sourceCode xml"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>&lt;<span class="kw">otherEntity</span>&gt;</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">entityName</span>&gt;site date&lt;/<span class="kw">entityName</span>&gt;</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">entityDescription</span>&gt;Field notes at site and date.&lt;/<span class="kw">entityDescription</span>&gt;</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">physical</span>&gt;</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">objectName</span>&gt;site_date.pdf&lt;/<span class="kw">objectName</span>&gt;</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">size</span><span class="ot"> unit=</span><span class="st">"byte"</span>&gt;9674&lt;/<span class="kw">size</span>&gt;</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">authentication</span><span class="ot"> method=</span><span class="st">"MD5"</span>&gt;8547b7a63fcf6c1f0913a5bd7549d9d1&lt;/<span class="kw">authentication</span>&gt;</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">dataFormat</span>&gt;</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">externallyDefinedFormat</span>&gt;</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>            &lt;<span class="kw">formatName</span>&gt;Portable Document Format&lt;/<span class="kw">formatName</span>&gt;</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>         &lt;/<span class="kw">externallyDefinedFormat</span>&gt;</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>      &lt;/<span class="kw">dataFormat</span>&gt;</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>   &lt;/<span class="kw">physical</span>&gt;</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">entityType</span>&gt;application/pdf&lt;/<span class="kw">entityType</span>&gt;</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>&lt;/<span class="kw">otherEntity</span>&gt;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The EML metadata should also include appropriate keywords describing the general purpose of the document or compressed archive (e.g., ice phenology, community composition, stream hydrology, etc.). For example, for still images, it is recommended to include keyword: image with the semantic annotation from the Information Artifact Ontology (IAO) :</p>
<pre><code>**Term IRI:** [http://purl.obolibrary.org/obo/IAO_0000101](http://purl.obolibrary.org/obo/IAO_0000101)


**Definition:** An image is an affine projection to a two dimensional surface, of measurements of some quality of an entity or entities repeated at regular intervals across a spatial range, where the measurements are represented as color and luminosity on the projected surface.</code></pre>
<p>Note, IAO includes at least one subcategory for image (e.g., <a href="http://www.ontobee.org/ontology/IAO?iri=http://purl.obolibrary.org/obo/IAO_0000185">photograph</a>). It is recommended the most specific applicable concept be used.</p>
</section>
<section id="data-inventory-table" class="level5" data-number="3.3.2.3.2">
<h5 data-number="3.3.2.3.2" class="anchored" data-anchor-id="data-inventory-table"><span class="header-section-number">3.3.2.3.2</span> Data Inventory Table</h5>
<p>We recommend that an additional level of metadata be provided through a data inventory table that effectively serves as a document catalog (see Table 4.1). The detail provided in this table should be guided by the same principles as stated above – to enable optimal usability of the documents. For example, still images from a stationary camera require latitude and longitude only in the EML file, not for each individual image. However, images from a moving camera may need that information for every image, or at least for every location (e.g., site, quadrat, transect). Additionally, Exif metadata from photographic images may be programmatically extracted to supplement the inventory table (refer to the <em>Tips and Tricks</em> section of <a href="data-gathered-with-Small-moving-platforms.html">Data Gathered with Small Moving Platforms</a>).</p>
<p>The data inventory table should be structured such that each column represents a particular attribute, described in EML as a <a href="https://eml.ecoinformatics.org/schema/eml-dataTable_xsd.html#eml-dataTable.xsd">dataTable</a> entity, and each row represents an individual document or a compressed archive of a group of documents. At minimum, the table should include an attribute for the document/archive filename, as well as any other essential attributes that vary per each document/archive. Additional attributes may include information on the date and/or time, but for this information to be useful, be consistent and use a controlled vocabulary for these fields so that a user can effectively search on them.</p>
<p>Table 1: Data inventory table structure.</p>
<table>
<tbody><tr>
<td>
<strong>Column</strong>
</td>
<td>
<strong>Attribute Description</strong>
</td>
</tr>
<tr>
<td>
Filename
</td>
<td>
Filename of each document or compressed archive, including file extension (e.g., ‘site_date.jpg’). For compressed archives, include the relative path of the document, with respect to the uncompressed directory structure (e.g., ‘2018/SITE3/quadrat4.jpg’).
</td>
</tr>
<tr>
<td>
Link/URL/URI
</td>
<td>
Link to download a document if it is available on a different system (also see <a href="data-in-other-repositories.html">Data in Other Repositories</a>). Persistent identifiers are recommended, if available.
</td>
</tr>
<tr>
<td>
Creator(s)
</td>
<td>
Name(s) of the creator(s) of the original document (e.g., photographer, field technician, interviewer). Multiple creators should be entered into a single cell using the pipe delimiter.
</td>
</tr>
<tr>
<td>
Datetime
</td>
<td>
Date (and time) associated with the document, in <a href="https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations">ISO 8601 format</a> (e.g., 2007-04-05T12:30-02:00).
</td>
</tr>
<tr>
<td>
Project specific datetime attributes
</td>
<td>
One or more appropriately labeled columns containing project specific date and time information for easier search and retrieval of documents (e.g., year, season, campaign).
</td>
</tr>
<tr>
<td>
Location
</td>
<td>
One or more location columns as appropriate, such as latitude and longitude in decimal degrees, site name, transect name, altitude, depth, habitat, etc.
</td>
</tr>
<tr>
<td>
Document specific attributes
</td>
<td>
One or more columns as appropriate to the document type, such as weather conditions, organism name, instrument type, etc.
</td>
</tr>
</tbody></table>
</section>
</section>
</section>
<section id="example-data-packages-in-edi-1" class="level3" data-number="3.3.3">
<h3 data-number="3.3.3" class="anchored" data-anchor-id="example-data-packages-in-edi-1"><span class="header-section-number">3.3.3</span> Example data packages in EDI</h3>
<p>Each of the Environmental Data Initiative (EDI) data packages listed below include images or other documents as data. Some of these packages contain data inventory tables (as dataTable entities) described in the EML metadata.</p>
<p>Table 2: Data packages in EDI providing examples of best practices from this document.</p>
<table>
<tbody><tr>
<td>
<strong>Dataset Title</strong>
</td>
<td>
<strong>Description</strong>
</td>
<td>
<strong>EDI Package ID</strong>
</td>
</tr>
<tr>
<td>
<em>Annual ground-based photographs taken at 15 net primary production (NPP) study sites at Jornada Basin LTER, 1996-ongoing</em>
</td>
<td>
Compressed archives of images grouped by year. Includes data inventory file.
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?scope=knb-lter-jrn&amp;identifier=210011005">knb-lter-jrn.210011005.105</a>
</td>
</tr>
<tr>
<td>
<em>McMurdo Dry Valleys LTER: Landscape Albedo in Taylor Valley, Antarctica from 2015 to 2019</em>
</td>
<td>
Compressed archives of aerial images, grouped by flight date, and associated reflectance data.
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?packageid=knb-lter-mcm.2016.2">knb-lter-mcm.2016.2</a>
</td>
</tr>
<tr>
<td>
<em>MCR LTER: Coral Reef: Computer Vision: Multi-annotator Comparison of Coral Photo Quadrat Analysis</em>
</td>
<td>
5090 coral reef survey images, and 251,988 random-point annotations by coral ecology experts.
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?scope=knb-lter-mcr&amp;identifier=5013">knb-lter-mcr.5013.3</a>
</td>
</tr>
<tr>
<td>
<em>Abundance and biovolume of taxonomically-resolved phytoplankton and microzooplankton imaged continuously underway with an Imaging FlowCytobot along the NES-LTER Transect in winter 2018</em>
</td>
<td>
144,281 images from a plankton imaging system with annotations and extracted size data.
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?packageid=knb-lter-nes.9.1">knb-lter-nes.9.1</a>
</td>
</tr>
<tr>
<td>
<em>Calling activity of Birds in the White Mountain National Forest: Audio Recordings (2016 and 2018)</em>
</td>
<td>
Compressed archive containing 410 audio files in wav format. Includes data inventory table.
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?scope=knb-lter-hbr&amp;identifier=268&amp;revision=1">knb-lter-hbr.268.1</a>
</td>
</tr>
</tbody></table>
</section>
<section id="resources-2" class="level3" data-number="3.3.4">
<h3 data-number="3.3.4" class="anchored" data-anchor-id="resources-2"><span class="header-section-number">3.3.4</span> Resources</h3>
<section id="considerations-for-digitizing-documents" class="level4" data-number="3.3.4.1">
<h4 data-number="3.3.4.1" class="anchored" data-anchor-id="considerations-for-digitizing-documents"><span class="header-section-number">3.3.4.1</span> Considerations for digitizing documents</h4>
<p>Following are some general considerations and recommendations for digitizing paper or other ‘hard-copy’ documents for archival. This is not meant to be an exhaustive list. For further and more detailed information, please refer to the U.S. National Archives and Records Administration (NARA)â€™s <em><a href="https://www.archives.gov/files/preservation/technical/guidelines.pdf">Technical Guidelines for Digitizing Archival Materials for Electronic Access</a></em>.</p>
<ul>
<li><strong>Effort.</strong> The decision to digitize documents, as well as the digitization method, involves trade-offs in the accessibility and ease of using particular hardware and/or software technologies, the quality of the digitization, and the overall effort spent. Digitization efforts may be significant, for example, when dealing with a large number of documents requiring meaningful file names, text recognition, and/or high resolution for improved accessibility.</li>
<li><strong>Equipment.</strong> Instruments for digitizing hard-copy documents range from high resolution scanners (less accessible, less user-friendly, more expensive, better quality) to smartphone cameras (ubiquitous, easy-to-use, lower quality). For example, taking a smartphone image in the field may be utilized for quick and easy digitization of field notes.</li>
<li><strong>Document resolution and file size.</strong> This is an important consideration that should be guided by the content and purpose of the document. Detailed paper maps should probably be scanned at high resolution and large file size, while field sheets may not need as much detail.</li>
<li><strong>Optical Character Recognition (OCR):</strong> When digitizing documents that include text, we recommend using scanning or other software with OCR capabilities (e.g., Adobe, ABBYY, Tesseract) to convert the text into machine readable characters so that the documents are searchable and thus, more usable. OCR does not work well for handwritten text, older fonts, or documents with busy backgrounds (speckled, dirty, faded, etc.).</li>
<li><strong>Sensitive Information and Human Subjects:</strong> Regardless of the digitization method, one should be mindful of sensitive information that shouldnâ€™t be archived or otherwise redacted (e.g., photographs of human subjects, field notebooks containing personal messages, gate combinations, and/or telephone numbers). In all cases in which human subjects are involved, Institutional Review Board (IRB) restrictions must be heeded. A signed IRB consent form for the associated research project represents a contract between researcher and human subject. It is important to note that IRB restrictions can differ among research studies within the same project. For further information, see the <a href="https://edirepository.org/about/edi-policy#sensitive-data">EDI Data Initiative Data Policy</a>.</li>
</ul>
<p>While transcription is a digitization method that can be performed on certain types of documents (e.g., audio/video recordings, field notebooks) and can enhance search capabilities, transcript generation requires substantially more effort than other digitization methods, and is prone to error. Moreover, in the case where the original documents contain drawings, transcripts may be incomplete or otherwise inaccurate. <em>Thus, we recommend digitizing documents by other means, using the considerations described above.</em></p>
</section>
</section>
</section>
<section id="spatial-data" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="spatial-data"><span class="header-section-number">3.4</span> Spatial Data</h2>
<p>Contributors: Tim Whiteaker, John Porter, Mary Martin</p>
<section id="introduction-3" class="level3" data-number="3.4.1">
<h3 data-number="3.4.1" class="anchored" data-anchor-id="introduction-3"><span class="header-section-number">3.4.1</span> Introduction</h3>
<p>This document/chapter contains recommendations on data package structure and metadata for spatial datasets. Over the timeline of Long Term Ecological Research (LTER) Network?s use of the Ecological Metadata Language (EML), both spatial data formats and data curation options have evolved. In this document, focus on best practices that can be widely adopted with the goal of enhancing data discoverability and usability, and the understanding that there are multiple solutions to creating these data packages.</p>
</section>
<section id="recommendations-for-data-packages-3" class="level3" data-number="3.4.2">
<h3 data-number="3.4.2" class="anchored" data-anchor-id="recommendations-for-data-packages-3"><span class="header-section-number">3.4.2</span> Recommendations for data packages</h3>
<section id="considerations-for-archiving-spatial-data" class="level4" data-number="3.4.2.1">
<h4 data-number="3.4.2.1" class="anchored" data-anchor-id="considerations-for-archiving-spatial-data"><span class="header-section-number">3.4.2.1</span> Considerations for archiving spatial data</h4>
<section id="data-formats" class="level5" data-number="3.4.2.1.1">
<h5 data-number="3.4.2.1.1" class="anchored" data-anchor-id="data-formats"><span class="header-section-number">3.4.2.1.1</span> Data formats</h5>
<p>To maximize reuse, avoid proprietary formats. The formats listed below can be read or imported by most mainstream GIS programs or with code using libraries such as GDAL.</p>
<p>Strongly recommended formats:</p>
<ul>
<li><strong>GeoTIFF</strong> - An open format for storing spatial raster data and metadata in a TIFF file.</li>
<li><strong>GeoPackage</strong> - A standard format from the Open Geospatial Consortium (OGC) for storing vector and raster data in a SQLite database file.</li>
</ul>
<p>Some other formats to consider are listed below.</p>
<ul>
<li><strong>KML/KMZ</strong> - Keyhole Markup Language (KML) file and its zipped version for storing vector data. This format was popularized by Google Earth and is now an OGC standard. KML is best visualized in Google software and may not render as well in other GIS software.</li>
<li><strong>GeoJSON</strong> - A format for storing vector data as text in JavaScript Object Notation (JSON). GeoJSON data are limited to the WGS84 coordinate system.</li>
<li><strong>netCDF/HDF5</strong> - binary formats originally designed for storing multidimensional arrays of spatial data typically organized onto a grid, but which now can accommodate vector data following the NetCDF Climate and Forecast Conventions (version 1.8 or higher).</li>
</ul>
<p>A couple of Esri formats are worth mentioning and are listed below.</p>
<ul>
<li><strong>File geodatabase</strong> - One of Esri’s formats for storing vector and raster information. Several feature classes and rasters can be stored in this folder and file based structure. GDAL’s OpenFileGDB driver enables non-Esri software to view at least the data layers in a file geodatabase, but usage of more advanced file geodatabase components such as topology rules or geometric networks may not be available outside of Esri software. Field types may not be imported correctly either. Export to GeoPackage instead, unless geodatabase is the only format that supports the advanced representation of your GIS data. Just know that you limit potential reuse of your data if you use this format.</li>
<li><strong>Shapefile</strong> - A legacy format for vector data which is widely supported. Be aware of <a href="https://en.wikipedia.org/wiki/Shapefile#Limitations">shapefile limitations</a> when considering this format. A shapefile consists of several individual files; include them as a single zip file in the data package. If the package has more than one shapefile, create a separate zip file for each shapefile.</li>
</ul>
<p>Although other open formats exist, their implementation in popular GIS software may be less common. If a proprietary format must be used to capture the full meaning of the data, consider also including a version of the data in an open format such as a simple data table along with metadata explaining its limitations in that format, or instructions on how to utilize the proprietary format. For example, an Esri layer package could be used when including recommended symbols for drawing vector features in a GIS is desired, in which case one could note that the vector data can be extracted by treating the layer package as a zip file.</p>
<p>Formats that are composed of more than one file, such as shapefiles, should be zipped. Include one dataset per zip file. For example, if you have 10 shapefiles, you would create 10 zip files.</p>
</section>
</section>
<section id="documenting-spatial-data-packages" class="level4" data-number="3.4.2.2">
<h4 data-number="3.4.2.2" class="anchored" data-anchor-id="documenting-spatial-data-packages"><span class="header-section-number">3.4.2.2</span> Documenting spatial data packages</h4>
<section id="document-as-spatialraster-vector-vs.-otherentity-in-eml" class="level5" data-number="3.4.2.2.1">
<h5 data-number="3.4.2.2.1" class="anchored" data-anchor-id="document-as-spatialraster-vector-vs.-otherentity-in-eml"><span class="header-section-number">3.4.2.2.1</span> Document as spatial[Raster, Vector] vs.&nbsp;otherEntity in EML</h5>
<p>There is a noticeable divergence in EDI spatial data packages, specifically, in the use of otherEntity vs spatial[Raster,Vector]. Here we discuss pros and cons of why one might choose to document spatial data with one type of EML entity over another. Either method is acceptable, and we recommend using spatial[Raster,Vector] when feasible. The documentation that follows provides best practices that will maximize discoverability and useability of spatial data, regardless of the entity type used.</p>
</section>
<section id="otherentity" class="level5" data-number="3.4.2.2.2">
<h5 data-number="3.4.2.2.2" class="anchored" data-anchor-id="otherentity"><span class="header-section-number">3.4.2.2.2</span> <a href="https://eml.ecoinformatics.org/schema/eml-dataset_xsd.html#DatasetType_otherEntity">otherEntity</a></h5>
<ul>
<li>Pros
<ul>
<li>EML preparation is simpler than with the spatial EML types</li>
<li>Allows aggregated data structures (e.g., file geodatabases)</li>
</ul></li>
<li>Cons
<ul>
<li>Spatial data stored as &lt;otherEntity&gt; might be harder to discover because it may be difficult to determine if data in an &lt;otherEntity&gt; is spatial data or some other type when searching or browsing</li>
<li>There is currently no controlled keywording to identify spatial data files that are included as otherEntity in EML</li>
<li>Tabular attributes of geometric entities may not be described in detail</li>
<li>Units (latitude/longitude vs meters vs feet) and projections may not be identified</li>
</ul></li>
</ul>
</section>
<section id="spatialrastervector" class="level5" data-number="3.4.2.2.3">
<h5 data-number="3.4.2.2.3" class="anchored" data-anchor-id="spatialrastervector"><span class="header-section-number">3.4.2.2.3</span> spatial[<a href="https://eml.ecoinformatics.org/schema/eml-dataset_xsd.html#DatasetType_spatialRaster">Raster</a>,<a href="https://eml.ecoinformatics.org/schema/eml-dataset_xsd.html#DatasetType_spatialVector">Vector</a>]</h5>
<ul>
<li>Pros
<ul>
<li>EML more fully describes vector attributes</li>
<li>There is a well documented path from Esri metadata to EML</li>
<li>An EML metadata search (on EDI or elsewhere) clearly identifies these as spatial datasets through the use of spatialRaster or spatialVector entities</li>
<li>LTER has built applications based on spatial[Raster,Vector] entities</li>
</ul></li>
<li>Cons
<ul>
<li>Data may not originate in ArcGIS, requiring a custom workflow to generate spatial entity EML</li>
<li>Spatial[Raster,Vector] can?t describe multi-layer aggregates of GIS data (e.g., geodatabases containing multiple feature classes)</li>
</ul></li>
</ul>
</section>
<section id="keywords" class="level5" data-number="3.4.2.2.4">
<h5 data-number="3.4.2.2.4" class="anchored" data-anchor-id="keywords"><span class="header-section-number">3.4.2.2.4</span> Keywords</h5>
<p>Clearly identifying a dataset as spatial in nature is important to discoverability. This can be achieved by the use of keywords in the EML keyword elements as well as in the title/abstract and methods where appropriate. Keywords frequently searched include: GIS, geographic information system, spatial data, plus the more specific format names like shapefile, geoTIFF etc. Consider including as appropriate.</p>
<p>Do include the keywords <strong>spatial vector</strong> and <strong>spatial raster</strong> as appropriate for your data. These keywords should be used especially if the data are archived as otherEntity.</p>
<p>You may also include keywords that describe broad spatial data layers, e.g., digital elevation model, elevation, boundary, land use, land cover, census, parcel, imagery, as well as keywords that describe the specifics associated with a broad spatial data layer, e.g., land cover types such as water and vegetation types, land use types such as urban and forest, and so on.</p>
</section>
<section id="gis-software-compatible-metadata" class="level5" data-number="3.4.2.2.5">
<h5 data-number="3.4.2.2.5" class="anchored" data-anchor-id="gis-software-compatible-metadata"><span class="header-section-number">3.4.2.2.5</span> GIS software compatible metadata</h5>
<p>GIS platforms will not ingest EML metadata. If your GIS software creates its own metadata file specific to that software, then it may be included as otherEntity. Be sure to populate this metadata, for example with descriptions and units for attributes in vector data or raster attribute tables. However, metadata in the standard ISO 19115 or CSDGM format to enable the metadata to be read by other GIS software is more useful.</p>
</section>
<section id="attribute-and-coordinate-system-detail-for-otherentity" class="level5" data-number="3.4.2.2.6">
<h5 data-number="3.4.2.2.6" class="anchored" data-anchor-id="attribute-and-coordinate-system-detail-for-otherentity"><span class="header-section-number">3.4.2.2.6</span> Attribute and coordinate system detail for otherEntity</h5>
<p>While the GIS software compatible metadata included in the package typically describes attributes and coordinate systems of the data, such descriptions should also be included in the EML metadata to help users determine fit for use prior to data download. The EML spatialVector and spatialRaster types include elements for this purpose. EML otherEntity can also include attribute descriptions; however, inclusion of attributes in this more generic element may not be as common, and the element does not formally support a description of coordinate systems.</p>
<p>When using <a href="https://eml.ecoinformatics.org/schema/eml-dataset_xsd.html#DatasetType_otherEntity">otherEntity</a> instead of spatialVector or spatialRaster, include coordinate system details in the otherEntity/entityDescription element. If not including a description of attributes in the otherEntity/attributeList element, at least include a summary description of attributes in otherEntity/entityDescription. If the spatial dataset and its associated metadata files are the only items in the data package, then you can include these descriptions in higher level EML elements such as the dataset abstract in addition to or in place of descriptions at the entity level.</p>
</section>
<section id="standardized-content-for-formats-and-entity-types" class="level5" data-number="3.4.2.2.7">
<h5 data-number="3.4.2.2.7" class="anchored" data-anchor-id="standardized-content-for-formats-and-entity-types"><span class="header-section-number">3.4.2.2.7</span> Standardized content for formats and entity types</h5>
<p>In EML physical/dataFormat/externallyDefinedFormat, include a <strong>formatName</strong> indicating the spatial data file format. We recommend using format names from the <a href="https://cn.dataone.org/cn/v2/formats">DataONE format list</a> when possible. Some spatial items from that list are shown below. Always check the list for the most up-to-date version of these names.</p>
<ul>
<li>Esri Shapefile (zipped)</li>
<li>Google Earth Keyhole Markup Language (KML)</li>
<li>Google Earth Keyhole Markup Language (KML) Compressed archive</li>
<li>Network Common Data Format, version 4</li>
<li>Hierarchical Data Format version 5 (HDF5)</li>
<li>GeoTIFF</li>
<li>GeoPackage Encoding Standard (OGC) Format Family</li>
<li>Esri File Geodatabase (zipped)</li>
<li>GeoJSON, version RFC 7946</li>
</ul>
<p>If your format is not included in the DataONE list, consider submitting an issue to that GitHub repository’s issue tracker so that the format can be added.</p>
<p>EML <strong>formatVersion</strong>, a sibling of formatName can be used to indicate the format version as in the example EML snippet below.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode xml code-with-copy"><code class="sourceCode xml"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>&lt;<span class="kw">externallyDefinedFormat</span>&gt;</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  &lt;<span class="kw">formatName</span>&gt;Network Common Data Format, version 4&lt;/<span class="kw">formatName</span>&gt;</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  &lt;<span class="kw">formatVersion</span>&gt;netCDF-4 classic&lt;/<span class="kw">formatVersion</span>&gt;</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>&lt;/<span class="kw">externallyDefinedFormat</span>&gt;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>For otherEntity, when populating the <strong>entityType</strong> element, use <strong>spatial raster</strong> or <strong>spatial vector</strong> as appropriate.</p>
</section>
</section>
</section>
</section>
<section id="data-gathered-with-small-moving-platforms" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="data-gathered-with-small-moving-platforms"><span class="header-section-number">3.5</span> Data Gathered with Small Moving Platforms</h2>
<p>Contributors: Sarah Elmendorf (lead), Tim Whiteaker, Lindsay Barbieri, Jane Wyngaard, Greg Maurer, Hap Garritt, Adam Sapp, Corinna Gries, Stace Beaulieu</p>
<section id="introduction-4" class="level3" data-number="3.5.1">
<h3 data-number="3.5.1" class="anchored" data-anchor-id="introduction-4"><span class="header-section-number">3.5.1</span> Introduction</h3>
<p>Modern advances in technology have increasingly allowed the collection of ecological data using small, often uncrewed, moving platforms. Systems variously known as small Uncrewed Aircraft Systems (sUAS), Uncrewed Surface Vehicles (USV), Autonomous, Uncrewed Underwater Vehicles (AUV or UUV) or ?drones,? more generally, now frequently serve as sensor carrying platforms. Moving platforms may also include gliders or animals with sensors affixed. Depending on the sensor(s) installed on the moving platform, data collected may include environmental measurements (temperature, concentration of chemicals), imagery (digital photos, multi- or hyperspectral sensors), or other remote-sensing acquisitions (ranging data, ground-penetrating radar). Example research applications include studies of vegetation cover and phenology, snowpack cover and depth, ground surface temperature, terrain elevation, bathymetry, species distribution or abundance, and many others.</p>
<p>Raw drone data can be voluminous and challenging to archive, but after processing, derived drone datasets typically resemble the more conventional spatial datasets that are regularly used in ecological research. In this document we focus on best practices for archiving raw and derived drone data, with particular attention to metadata and processing code that is specific to drone datasets. Note that this chapter does not specifically address data collected by large moving platforms like airplanes and satellites, or by human and animal platforms.</p>
</section>
<section id="recommendations-for-data-packages-4" class="level3" data-number="3.5.2">
<h3 data-number="3.5.2" class="anchored" data-anchor-id="recommendations-for-data-packages-4"><span class="header-section-number">3.5.2</span> Recommendations for data packages</h3>
<p>General considerations for archiving data from moving sensor platforms</p>
<ul>
<li><strong>Repository</strong>: We are currently unaware of many specialized repositories for these data, and therefore, EDI is used as the representative data repository for many cases presented here. Repositories other than EDI may have specific metadata formatting requirements, but the general recommendations with regard to content could presumably be applied. For LiDAR based UAV data, consider contributing to Open Topography <a href="https://opentopography.org/">https://opentopography.org/</a>); for AUV data, the U.S. Marine Geoscience Data System (MGDS) (<a href="http://www.marine-geo.org/index.php">http://www.marine-geo.org/index.php</a>) which serves the IEDA “MGDL” node in DataONE is a good option. Glider data may be contributed to the U.S. IOOS Glider DAC (<a href="https://slack-redir.net/link?url=https%3A%2F%2Fgliders.ioos.us%2Fdata%2F&amp;v=3">https://gliders.ioos.us/data/</a>), archived at the National Centers for Environmental Information (NCEI) thus fulfilling NSF OCE Data Policy. If a decision is made to archive an LTER drone dataset in an external (i.e.&nbsp;non-EDI) repository but links to EDI data packages are desired, recommendations in the <a href="data-in-other-repositories.html">Data in Other Repositories</a> chapter may apply.</li>
<li><strong>Size of data set</strong>: The file size of raw data from drone imagery can be substantial. If large volumes of raw data (&gt;100 GB in total) are to be archived on EDI, please coordinate with EDI and follow the best practices for <a href="large-data-sets.htm">Large Data Sets</a>. Even if raw data are in a proprietary binary format and specific software is required for processing, publishing them may be important for reprocessing when software improves.</li>
<li><strong>Designing a data package</strong>: In many applications of moving sensor data, raw images/measurements must be processed to arrive at data products that can be analyzed to answer research questions. To enable a fully reproducible analysis pipeline, we recommend archiving three components: the raw data, any key derived data products (e.g., orthomosaic images, DEMs, DSMs, NDVI, landcover, snow depth, or surface temperature maps), and the processing code. These three components may be archived in separate data packages or together, and each should follow accepted best practices for its data type. To archive raw image collections, for example, see the considerations on grouping images into compressed archives (.zip, .tar) and creating an inventory file, as described in the <a href="images-and-documents-as-data.html">Images and Documents as Data</a> chapter. For derived geospatial files, such as DEMs, refer to the <a href="spatial-data.htm">Spatial Data</a> chapter. Custom processing code should be archived with the data following recommendations in the <a href="code.html">Code in EDI</a> chapter. If a standalone program is used to process data, reference the program in the methods metadata with adequate details to ensure reproducibility (name, version, date, configuration, etc.).</li>
</ul>
<section id="metadata-for-moving-platform-data-packages-at-edi" class="level4" data-number="3.5.2.1">
<h4 data-number="3.5.2.1" class="anchored" data-anchor-id="metadata-for-moving-platform-data-packages-at-edi"><span class="header-section-number">3.5.2.1</span> Metadata for moving platform data packages at EDI</h4>
<p>The data package should include metadata elements that, at a minimum, (a) identify it as being collected by a moving platform, (b) deliver basic information about the data collection platform, instrument payload (camera, sensors), and procedure (flight information or similar), and (c) deliver necessary information about post-processing of the raw camera or sensor data, if any. Accordingly, these recommendations vary based on whether the data package contains raw or derived data.</p>
<p>High level metadata pertaining to the entire data package are easily provided in the EML file (e.g.&nbsp;a geographic bounding box). Data packages from drones or other moving platforms commonly include numerous point measurements, images, or other granular data entities, either separately or inside a compressed archive file. Detailed metadata pertaining to these data entities may be included as additional files in the data package. Inventory tables, usually a simple CSV file, are one such additional metadata format. For example, an inventory table could be used to list individual data files in the data package (e.g., images from one drone flight) and provide metadata (e.g.&nbsp;point location) about each. In addition to inventory tables, files that enable or supplement common processing pipelines, such as flight or mission logs, may be included. A flight/mission log may be provided in a proprietary binary format, but because software for parsing these formats may become obsolete, we recommend archiving the log in the format most useful for contemporary analysis software, and extracting and appending the information to the inventory file where appropriate. Exif (Exchangeable image file format) metadata in images may also be programmatically extracted to supplement the inventory file.</p>
<p>Clearly, there are many possible ways to combine raw data, derived data, and metadata files into a moving platform data package. No matter the combination, the critical metadata categories and the recommended contents below should be considered and included where possible. The decision on whether to provide the metadata in EML or at a more granular level, such as an inventory table, will depend on the given dataset.</p>
<ul>
<li><strong>Methods:</strong> unique identifier for a given flight or mission; summary information from a flight log; weather conditions; accuracy of sensor and geographic location information; data processing method; ground sample distance; image overlap; flight height; whether UAS followed terrain elevation vs fixed-height flight; location of UAS launch (since some image metadata are derived from this); general description of software used and for what purpose; sensor calibration date and procedure; general description of payload type, such as multispectral camera and spectral bands.</li>
<li><strong>Instrumentation:</strong> make and model of platform, sensor, and camera, including manufacturer and specific model names and numbers. Include make and model of any interchangeable lenses in cameras. Specifics like spectral bands, temperature range, sensor accuracy, etc.</li>
<li><strong>Software:</strong> (see also <a href="code.html">Code in EDI</a> chapter) list of software used. Especially when code is proprietary or archived elsewhere, the name, version, and configuration of any software used are advisable, as are corrections applied (e.g., correction for sensor angle or heat/air flow). Ideally, a .pdf report generated by processing software can be archived as an otherEntity together with the imagery itself to convey much of the necessary information. Also include data used as a ground truth or calibration points for post processing (e.g.&nbsp;spectral calibration image/biomass sample/wind speed/etc) and their date of collection.</li>
<li><strong>People with specified role:</strong> drone operator, image processor</li>
<li><strong>Geographic Information:</strong> (see also <a href="sptial-data.html">Spatial Data chapter</a>) a general bounding box should be included in EML, while the individual location of images or point measurements should be handled in the inventory table, or directly in the included data files. Also include the coordinate reference system (e.g., WGS 84) used for images and (if different) ground control points, projection type if needed, altitude of image/measurement acquisition, spatio-temporal coordinates, pitch, roll, and yaw from flight log or image data points. It should be noted that there are special considerations for underwater vehicles, especially with regard to metadata to explain how geographic positions were obtained. With autonomous underwater vehicles (AUVs), there can be error sources in the topside GPS localization, the underwater acoustic positioning system (e.g., Long Baseline, Ultrashort Baseline), as well as any sensors used for dead reckoning (e.g., accelerometer, Doppler Velocity Log). At a minimum, it would be useful to know which sensors were used to produce the localization data and whether the navigation tracklines were post-processed with benchmarks.</li>
<li><strong>Temporal Information:</strong> may also be provided at the EML level or as timestamps for individual data/image points, either in inventory tables or in the data files themselves. Time of day critically affects useability for image-based datasets; so ensure that the time of day is clear from the metadata available prior to download, either in the EML temporal coverage or via the methods.</li>
<li><strong>Keywords:</strong> Use of appropriate keywords aids in data discovery. Keywords that identify datasets as drone-related are therefore recommended (e.g.&nbsp;drone, UAV, UAS). Keywords describing the type of data collected are also recommended (e.g.&nbsp;image collection, aerial imagery, thermal imagery, NDVI, digital elevation map). For drone mapping data products, keyword recommendations from the <a href="spatial-data.html">Spatial Data chapter</a> are largely applicable.</li>
</ul>
</section>
<section id="examples-and-additional-metadata-guidance" class="level4" data-number="3.5.2.2">
<h4 data-number="3.5.2.2" class="anchored" data-anchor-id="examples-and-additional-metadata-guidance"><span class="header-section-number">3.5.2.2</span> Examples and additional metadata guidance</h4>
<p>Several EDI data packages for data from moving platforms are presented as examples in Table 1. Many more detailed, ?drone-specific? metadata terms and values can be included in data packages for drones and other moving platforms. For completeness we have developed a comprehensive list of recommended and optional metadata terms based on the work of Wyngaard et al.&nbsp;(2019), Thorner et al.&nbsp;(2020), with mappings to select relevant ontologies, viewable <a href="https://docs.google.com/spreadsheets/d/1PQ0SUEQLgQXdz2PUNDy2jGry3o9veAedcz8l-5ubpwU">here</a>. For each metadata element, we assessed its utility in terms of data discovery, evaluating fitness for use, and actual data reuse. The minimum recommended subsets of metadata that are included in the section above were derived from this table.</p>
<p>Table 1. Example packages at EDI and other repositories</p>
<table>
<tbody><tr>
<td>
<strong>Title</strong>
</td>
<td>
<strong>Description</strong>
</td>
<td>
<strong>EDI packageID</strong>
</td>
</tr>
<tr>
<td>
<em>Orthophoto and elevation models from UAV overflights at the G-IBPE study site at Jornada Basin LTER in 2019</em>
</td>
<td>
Approximately 599 RGB images and data derived from uncrewed aerial vehicle (UAV) overflights of the G-IBPE study site at the Jornada Basin LTER in southern New Mexico, USA.
</td>
<td>
<a href="https://portal-s.edirepository.org/nis/mapbrowse?scope=knb-lter-jrn&amp;identifier=210543001">knb-lter-jrn.210543001</a>
</td>
</tr>
<tr>
<td>
<em>Aerial imagery from unmanned aerial systems (UAS) flights and ground control points: Plum Island Estuary and Parker River NWR (PRNWR), February 27th, 2018.</em>
</td>
<td>
USGS Aerial imagery UAS flights at the Parker River National Wildlife Refuge, Massachusetts, USA, includes ground control, multispectral and true color child items which each have data entities that include ground control or a file catalog of images
</td>
<td>
<a href="https://www.sciencebase.gov/catalog/item/5c0fe16de4b0c53ecb2d1bc3">ScienceBase</a>
</td>
</tr>
<tr>
<td>
<em>Spatial variability in water chemistry of four Wisconsin aquatic ecosystems - High speed limnology Environmental Science and Technology datasets</em>
</td>
<td>
water chemistry sensors embedded in a high-speed water intake system to document spatial variability.
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?packageid=knb-lter-ntl.337.4">knb-lter-ntl.337.4</a>
</td>
</tr>
<tr>
<td>
<em>Thermal infrared, multispectral, and photogrammetric data collected by drone for hydrogeologic analysis of the East River beaver-impacted corridor near Crested Butte, Colorado</em>
</td>
<td>
infrared, multispectral, visual image data, and derivative products (orthomosaic and digital surface model) collected along a beaver-impacted section of the East River from August 12-17, 2017 and July 28-August 2, 2018.
</td>
<td>
<a href="https://www.sciencebase.gov/catalog/item/5ccc4cc9e4b09b8c0b78c97a">ScienceBase</a>
</td>
</tr>
</tbody></table>
</section>
</section>
<section id="resources-3" class="level3" data-number="3.5.3">
<h3 data-number="3.5.3" class="anchored" data-anchor-id="resources-3"><span class="header-section-number">3.5.3</span> Resources</h3>
<section id="tips-and-tricks" class="level4" data-number="3.5.3.1">
<h4 data-number="3.5.3.1" class="anchored" data-anchor-id="tips-and-tricks"><span class="header-section-number">3.5.3.1</span> Tips and Tricks</h4>
<p>For making an image catalog (.csv) from a directory of images, consider using the exif tool <a href="https://exiftool.org/">https://exiftool.org/</a>. For example, the command ?exiftool.exe -csv -r mydirectory &gt; image_catalog.csv? will extract the entirety of the exif tags from all files stored under mydirectory into a comma-delimited table and write it to the the file image_catalog.csv</p>
</section>
<section id="semantic-annotation" class="level4" data-number="3.5.3.2">
<h4 data-number="3.5.3.2" class="anchored" data-anchor-id="semantic-annotation"><span class="header-section-number">3.5.3.2</span> Semantic Annotation</h4>
<p>Semantic annotation of drone imagery is a rapidly developing field. Ontologies that provide relevant terms include: <a href="http://www.dronetology.net/">dronetology</a>; <a href="http://www.sensorml.com/ontologies.html">sensorML</a>; <a href="https://www.fgdc.gov/metadata/csdgm/">FGDC content standard for digital geospatial metadata</a> (not officially an ontology but a structured metadata format with defined terms); <a href="https://www.w3.org/TR/vocab-ssn/">Semantic Sensor Network ontology</a> (SSN, including the SOSA core); <a href="https://bioportal.bioontology.org/ontologies/SWEET">Semantic Web for Earth and Environment Technology ontology</a> (SWEET); and <a href="http://www.obofoundry.org/ontology/envo.html">Environment Ontology</a>.</p>
</section>
<section id="references" class="level4" data-number="3.5.3.3">
<h4 data-number="3.5.3.3" class="anchored" data-anchor-id="references"><span class="header-section-number">3.5.3.3</span> References</h4>
<p>Thomer, Andrea K., Swanz, Sarah, Barbieri, Lindsay, Wyngaard, Jane. (2020). A minimum information framework the FAIR collection of earth and environmental science data with drones. DOI: 10.5281/zenodo.4017647</p>
<p>Wyngaard, J.; Barbieri, L.; Thomer, A.; Adams, J.; Sullivan, D.; Crosby, C.; Parr, C.; Klump, J.; Raj Shrestha, S.; Bell, T. Emergent Challenges for Science sUAS Data Management: Fairness through Community Engagement and Best Practices Development. <em>Remote Sens.</em> <strong>2019</strong>, <em>11</em>, 1797.</p>
</section>
</section>
</section>
<section id="data-in-other-repositories" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="data-in-other-repositories"><span class="header-section-number">3.6</span> Data in Other Repositories</h2>
<p>Contributors: Greg Maurer (lead), Stace Beaulieu, RenÃ©e Brown, Sarah Elmendorf, Hap Garritt, Gastil Gastil-Buhl, Corinna Gries, Li Kui, An Nguyen, John Porter, Margaret O’Brien, Tim Whiteaker</p>
<section id="introduction-5" class="level3" data-number="3.6.1">
<h3 data-number="3.6.1" class="anchored" data-anchor-id="introduction-5"><span class="header-section-number">3.6.1</span> Introduction</h3>
<p>A wide variety of data repositories are available for publishing biological, environmental, and Earth observation data, and the choice of where to publish a particular dataset is determined by many competing factors. For example, a funding agency or journal may require a certain repository (e.g., NSF BCO-DMO, NSF ADC, USDA ADC, DOE ESS-DIVE); the research subject or data type may be best served by a specialized repository (e.g., AmeriFlux, GenBank); or datasets may be submitted to a general purpose repository with minimal metadata requirements to simplify and speed data publishing (e.g., DRYAD, Figshare, Zenodo). For these and other reasons related datasets are sometimes published in disparate data repositories, the same data needs to be discoverable in more than one repository, or multiple datasets from one or more repositories may be used to create a new, derived dataset. In such cases, it can be advantageous to establish links between datasets in different repositories such that provenance, supplementation, duplication or other relationships are explicit. Clearly, this subject goes well beyond the single repository and better standards and approaches for linking resources and documenting data provenance are being developed elsewhere (e.g.&nbsp;<a href="https://www.dataone.org/network/">DataONE</a>, <a href="https://purl.dataone.org/provone-v1-dev">ProvONE</a>, <a href="https://wholetale.org/">WholeTale</a>). Here we concentrate on specific cases in the context of large and multidisciplinary projects, such as LTER sites, that wish to enhance data discovery and preserve data relationships across multiple repositories.</p>
</section>
<section id="recommendations-for-data-packages-5" class="level3" data-number="3.6.2">
<h3 data-number="3.6.2" class="anchored" data-anchor-id="recommendations-for-data-packages-5"><span class="header-section-number">3.6.2</span> Recommendations for data packages</h3>
<section id="considerations-for-creating-linked-data" class="level4" data-number="3.6.2.1">
<h4 data-number="3.6.2.1" class="anchored" data-anchor-id="considerations-for-creating-linked-data"><span class="header-section-number">3.6.2.1</span> Considerations for creating linked data</h4>
<p>In practice, links to data in other repositories can be achieved using metadata only, by including a data inventory file, or, although not recommended, by duplicating the data in the new repository record. <strong>Generally, duplicating data in multiple repositories is not recommended because it creates two problems.</strong> First, it is a burden to maintain multiple copies of a dataset and avoid divergence between them. Second, it can create confusion for data re-users who may download or cite the same data multiple times. Care must be taken to clearly identify such duplications for data users when they are created. Whenever linked datasets are created, it is strongly recommended that both repositories are aligned with FAIR data principles, <a href="https://doi.org/10.1038/sdata.2016.18">outlined here</a>, so that users have unfettered access to all data and metadata.</p>
<p>In addition to these considerations, there are a number of reasons to create a new repository record that is linked to data in other repositories. Each of these reasons, which are outlined below, has pros and cons that will need to be weighed from the different perspectives of the data user, data provider and research project management requirements.</p>
<ul>
<li><strong>Requirements dictate multiple repositories:</strong> Large research projects or sites are frequently funded by different agencies and programs. Data collection may be supported by several such funding streams and, hence, fall in the purview of more than one requirement to archive data in a particular repository. In some cases, data repositories already accommodate such requirements by linking or replicating data appropriately. Examples of this are LTER data in EDI, NSF BCO-DMO and NSF ADC.</li>
<li><strong>Adding important metadata</strong>: If data were originally submitted to a general purpose repository with minimal metadata requirements (e.g., DRYAD, figshare) additional metadata (e.g., EML) may be needed for discoverability, reusability, and integration. By creating a new repository record that identifies and is linked to the original published dataset, richer and more useful metadata can be added to the new record and utilized.</li>
<li><strong>Use of specialist repositories for related data:</strong> There are sometimes advantages to publishing particular data types in specialized repositories. Specialized data repositories (e.g.&nbsp;GenBank, AmeriFlux) usually enforce strict data formatting, provide quality standards, enhanced search, discovery and reuse of particular types of data across projects in a way that is not possible using a generalized metadata format (EML) and repository (EDI). However, these data may not be discoverable with other, related project data taken at the same location and time. Creating links between related datasets held in specialist and generalist repositories helps preserve this context.</li>
<li><strong>A derived data product is archived in a different repository than the source (raw) data:</strong> A wide range of cases fall into this category, from a direct one to one relationship of, e.g., a gene sequence and its OTU identification, a metagenome analysis and its community diversity metrics, to several datasets being combined in synthesis or meta-analysis studies. In these cases, links between source data and derived data products that are published in separate repositories need to be created and clearly documented.</li>
<li><strong>Linking to site- or project-relevant data from other research groups or agencies:</strong> Although it may help with some aspects of data discovery it is generally not recommended to create records in EDI for data collected and managed by entirely different research groups or agencies. <strong><em>In these cases, however, it is recommended to place a pointer to such repositories on a project website or develop other means for data users to discover relevant resources.</em></strong></li>
</ul>
</section>
<section id="general-metadata-for-linked-data-packages-in-edi" class="level4" data-number="3.6.2.2">
<h4 data-number="3.6.2.2" class="anchored" data-anchor-id="general-metadata-for-linked-data-packages-in-edi"><span class="header-section-number">3.6.2.2</span> General metadata for linked data packages in EDI</h4>
<p>In EDI, the linked data package can be assembled using standard practices and EML metadata elements, but the included metadata and data entities must clearly lead the data user to files held in outside repositories. In addition, the package metadata should communicate the essential elements needed for data discovery (subject matter, authors, location, time-frame, etc.) and a brief description of how the data may be accessed, re-used, and cited via the outside repository as needed. General guidance on the content and structure of key metadata elements in an EDI data package linked to data in other repositories are described below.</p>
<ul>
<li><strong>Abstract:</strong> Describe the key features of the data package. If the data package contains only links to data held in other repositories, or data duplicated from another repository, clearly state that the original data are located in a different repository and direct the user to the correct data citation. Describe the target data in sufficient detail that users can determine whether these data are fit for their use, and instruct them on how to find and re-use the data.</li>
<li><strong>Methods:</strong> Collection/generation methods for any data entities included or linked to. If the methods are well-described in the metadata at another repository, this element can simply refer users there. If the new data package includes ancillary data or derived data, describe how those data were collected or derived.</li>
<li><strong>Geographic description</strong> and <strong>coordinates:</strong> At a minimum these elements should define a bounding box that will make the data package discoverable through EDI, DataOne, or other geographic search interfaces. Additional, more detailed coordinates may be given in the inventory file entity as described below.</li>
<li><strong>Keywords:</strong> Since some linked data packages include an inventory of data held at a different repository, include the keyword “data inventory” and thematic keywords that describe the data entities in the other repository.</li>
</ul>
</section>
<section id="common-use-cases-and-their-structure-in-edi" class="level4" data-number="3.6.2.3">
<h4 data-number="3.6.2.3" class="anchored" data-anchor-id="common-use-cases-and-their-structure-in-edi"><span class="header-section-number">3.6.2.3</span> Common use cases and their structure in EDI</h4>
<p>There are several common use cases for creating a new linked data package in EDI. The new package may establish either a one-to-one link from EDI to a dataset in another repository, or a one-to-many relationship that is more complex. Three possible cases are described below in terms of what entities to publish, where to publish them, metadata elements to be created in EML, and the contents of included data entities. There are likely to be other use cases for linking EDI data packages to other repositories.</p>
<p><strong>Case 1:</strong> One dataset needs to be discoverable in more than one data repository. The data remain the same, but the metadata in the new data package at EDI may be upgraded beyond what exists in the other repository.</p>
<ul>
<li>The metadata in EDI must clearly state, preferably in the abstract or another obvious location, that this data package is already published in another repository. Include the original unique identifier and instruct users to cite that original data, if appropriate.<br>
</li>
<li>Include instructions on how to access and cite the original data if the original repository is lacking in such guidance.</li>
<li>If data are duplicated (which is not recommended), metadata should include information on how versions in different repositories are kept synchronized. If such synchronization is not feasible, users should be warned to inspect both sources for the latest data..</li>
<li>In EML the &lt;additionalIdentifier&gt; field may be used to store the persistent identifier (DOI), or a link (URL) that refers to the data held in another repository to make the link machine readable. Where an external repository supplies both a URL and DOI, use the DOI as URLs may not be maintained through time.</li>
</ul>
<p><strong>Case 2:</strong> A list of data records held in a specialized repository needs to be linked to ancillary or supporting data that are being published in EDI (for derived data see Case 3).</p>
<ul>
<li>This case applies when a collection of datasets, or similar scientific resources, is held in a specialized repository and closely related ancillary or supporting data and metadata needs to be archived in a more generalist data repository like EDI. For example, ancillary environmental data or laboratory analyses held in EDI could be linked to collections of sequence reads held in NCBI GenBank or museum voucher specimens archived with Darwin Core metadata. See complete examples in Table 5.3.</li>
<li>The new EDI data package should include a ‘data inventory’ (or manifest of holdings) file as a data entity. This is most likely a simple tabular data file, such as a CSV, that lists and describes the repository records held in the specialized data repository and has its column attributes described in EML as a <a href="https://eml.ecoinformatics.org/schema/eml-dataTable_xsd.html#eml-dataTable.xsd">dataTable</a> entity.</li>
<li>The inventory table must have a row for each outside repository record (or some meaningful grouping of records, e.g., project in NCBI) being linked to, with columns that include persistent unique identifiers of the data in the other repository, and relevant descriptors of the data. The complete content of the inventory will be dictated by the structure of the other repository and the data entities and metadata held there. Suggested columns are presented in Table 5.1.</li>
<li>The inventory table may also provide additional contextual information for each individual data resource in another repository. Table 5.2 presents examples of these contextual columns. They are, however, subject dependent and may vary for different projects. For more examples, see the discussion on sequencing and genomic data later in this document.</li>
</ul>
<p>Table 1: Suggested columns for identifying the external data in the data inventory table.</p>
<table>
<tbody><tr>
<td>
<strong>Column</strong>
</td>
<td>
<strong>Description</strong>
</td>
</tr>
<tr>
<td>
External unique ID
</td>
<td>
Unique identifier for the data resource in the other repository. E.g. Accession number
</td>
</tr>
<tr>
<td>
External access URL
</td>
<td>
A unique, persistent link to the data resource in the other repository.
</td>
</tr>
<tr>
<td>
Title/description
</td>
<td>
Title and/or brief description of the data resource
</td>
</tr>
<tr>
<td>
Filename(s)
</td>
<td>
Dataset or file name at the other repository
</td>
</tr>
<tr>
<td>
Format
</td>
<td>
File format of above
</td>
</tr>
<tr>
<td>
Repository URL
</td>
<td>
URL of the repository being linked to
</td>
</tr>
</tbody></table>
<p>Table 2: Examples of additional contextual columns in the data inventory table.</p>
<table>
<tbody><tr>
<td>
<strong>Column</strong>
</td>
<td>
<strong>Description</strong>
</td>
</tr>
<tr>
<td>
Latitude/Longitude
</td>
<td>
Latitude and longitude in standard format for each data resource in the other repository.
</td>
</tr>
<tr>
<td>
Location name
</td>
<td>
Locally used name of collection site
</td>
</tr>
<tr>
<td>
Treatment level
</td>
<td>
Experimental treatment applied to the outside dataset
</td>
</tr>
<tr>
<td>
Start/End datetime
</td>
<td>
Starting/ending datetime of the data resource (NA for End if data collection is ongoing)
</td>
</tr>
<tr>
<td>
Reference publication
</td>
<td>
DOI of publication providing in-depth context for data
</td>
</tr>
</tbody></table>
<p><strong>Case 3:</strong> One or more datasets in other repositories are used to create derived data products that need to be archived in EDI.</p>
<ul>
<li>In this case the new dataset is directly or indirectly derived from the ‘source’ dataset(s) in other repositories. Such derived data may serve a wide range of research purposes, including use in cross-site synthesis, re-analysis, or meta-analysis studies.</li>
<li>Provenance metadata should be used to describe the relationship between the source and derived datasets, which ensures reproducibility and preserves data lineage. In a new EDI data package that archives derived data, the provenance metadata should be inserted in the EML file utilizing &lt;dataSource&gt; elements. The &lt;dataSource&gt; elements should be nested within a &lt;methodStep&gt; element and will establish the links to any source datasets located in another repository. An example snippet of provenance EML is shown in Figure 1.</li>
<li>Other cross-repository standards for provenance metadata are still being developed and are not widely adopted, e.g., <a href="http://homepages.cs.ncl.ac.uk/paolo.missier/doc/dataone-prov-3-years-later.pdf">ProvONE</a>.</li>
<li>The EDI portal interface provides <a href="https://portal.edirepository.org/nis/provenanceGenerator.jsp">automatic generation of provenance metadata</a> EML snippets for datasets in EDI. The <a href="https://github.com/EDIorg/emlAssemblyLine">EMLassemblyline</a> and <a href="https://github.com/BLE-LTER/MetaEgress">MetaEgress</a> (in connection with <a href="https://github.com/lter/LTER-core-metabase">LTER-core-metabase</a>) R packages for EML creation will also generate provenance metadata.</li>
</ul>
<p>Example 1: EML snippet with a data provenance methodStep:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode xml code-with-copy"><code class="sourceCode xml"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>&lt;<span class="kw">methodStep</span>&gt;</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">description</span>&gt;</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">para</span>&gt;This methodStep contains data provenance information as specified in the LTER EML Best Practices. Each dataSource element here lists entity-specific information and links to source data used in the creation of this derivative data package.&lt;/<span class="kw">para</span>&gt;</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>   &lt;/<span class="kw">description</span>&gt;</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>   &lt;<span class="kw">dataSource</span>&gt;</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">title</span>&gt;Source dataset title&lt;/<span class="kw">title</span>&gt;</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">creator</span>&gt;</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">individualName</span>&gt;</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>            &lt;<span class="kw">givenName</span>&gt;first name&lt;/<span class="kw">givenName</span>&gt;</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>            &lt;<span class="kw">surName</span>&gt;last name&lt;/<span class="kw">surName</span>&gt;</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>         &lt;/<span class="kw">individualName</span>&gt;</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">organizationName</span>&gt;organization name&lt;/<span class="kw">organizationName</span>&gt;</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">electronicMailAddress</span>&gt;email@some.edu&lt;/<span class="kw">electronicMailAddress</span>&gt;</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>      &lt;/<span class="kw">creator</span>&gt;</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">distribution</span>&gt;</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">online</span>&gt;</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>            &lt;<span class="kw">onlineDescription</span>&gt;This is a link to an external online data resource (describe resource and repository location).&lt;/<span class="kw">onlineDescription</span>&gt;</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>            &lt;<span class="kw">url</span><span class="ot"> function=</span><span class="st">"information"</span>&gt;https://pasta.lternet.edu/package/metadata/eml/knb-lter-ntl/80/2&lt;/<span class="kw">url</span>&gt;</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>         &lt;/<span class="kw">online</span>&gt;</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>      &lt;/<span class="kw">distribution</span>&gt;</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">contact</span>&gt;</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">positionName</span>&gt;Information Manager&lt;/<span class="kw">positionName</span>&gt;</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">organizationName</span>&gt;organization name&lt;/<span class="kw">organizationName</span>&gt;</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>         &lt;<span class="kw">electronicMailAddress</span>&gt;infomgr@some.edu&lt;/<span class="kw">electronicMailAddress</span>&gt;</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>      &lt;/<span class="kw">contact</span>&gt;</span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>   &lt;/<span class="kw">dataSource</span>&gt;</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>&lt;/<span class="kw">methodStep</span>&gt;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="nucleotide-sequence-and-genomic-data" class="level3" data-number="3.6.3">
<h3 data-number="3.6.3" class="anchored" data-anchor-id="nucleotide-sequence-and-genomic-data"><span class="header-section-number">3.6.3</span> Nucleotide sequence and genomic data</h3>
<p>Nucleotide sequence data consists of the order and arrangement of DNA or RNA bases extracted from individual organisms or environmental samples. Similarly, genomic data refers to the complete genetic information (either DNA or RNA) of an organism, while metagenomic data refers to the study of genomes recovered from environmental samples. Sequencing, genomic and metagenomic datasets can be very large and complex, and researchers in these fields benefit from particular methods of data access, analysis, and collaboration. Therefore, these data have specialized requirements for data archiving.</p>
<p>Archiving nucleotide sequence and genomic (or other ‘<a href="https://en.wikipedia.org/wiki/Omics">omics</a>’) data are a common use case for creating linked datasets. Data that originate from nucleotide sequencing techniques are most often stored in specialized repositories such as National Center for Biotechnology Information (NCBI) GenBank and the European Nucleotide Archive. However, while sequences or assembled genomes constitute important raw data, ancillary and derived data products related to these raw data are frequently published in repositories specializing in ecological data. For example, data derived from sequence data, such as operational taxonomic units (OTUs) or functional assignments, and ancillary data that describe the environmental, biochemical, or experimental context of the sequencing data, are often included in scientific publications, and do not always fit within the scope of a specialized sequence or genome data repository.</p>
<section id="recommendations-for-sequencing-or-genomic-datasets" class="level4" data-number="3.6.3.1">
<h4 data-number="3.6.3.1" class="anchored" data-anchor-id="recommendations-for-sequencing-or-genomic-datasets"><span class="header-section-number">3.6.3.1</span> Recommendations for sequencing or genomic datasets</h4>
<p>Linking to genomics data is an example of Case 2 described above. Summaries or inventories of data records held in a repository like NCBI GenBank are linked to their derived products or additional measurements published in a more generalist repository such as EDI.</p>
<p>In addition to the metadata typically included with any data package published by the site or research group, include metadata that is descriptive specifically of sequencing and genomics datasets. It is recommended to refer to the <a href="https://press3.mcs.anl.gov/gensc/mixs/">MixS templates</a> for standard terminology, especially in the keyword section:</p>
<p><strong>Keywords</strong> that can help users discover the sequencing or genomic dataset include:</p>
<ol type="1">
<li>General data type descriptions (‘nucleotide sequence’, ‘genomics’, ‘metagenomics’)</li>
<li>Names of target genes or subfragments (‘16S rRNA’, ‘18S rRNA’, ‘nif’, ‘amoA’, ‘rpo’, ‘ITS’)</li>
<li>Names of the sequencing technique (‘Sanger’, ‘pyrosequencing’, ‘ABI-solid’)</li>
<li>Names of the linked repository (‘SRA’, ‘EMBL’, ‘Ensembl’)</li>
<li>Descriptors of included ancillary data (‘nitrogen’, ‘soil’, ‘drought’)</li>
<li>Descriptors of derived data products (‘OTU’, ‘functional annotation’, ‘population’)</li>
</ol>
<p><strong>Inventory tables</strong> are of central importance to datasets that index data resources in a sequencing or genomics repository. It is recommended that this inventory should have the columns described in Table 1. Note that the unique identifiers included will depend on the granularity of the links to the outside repository. For example, in NCBI, there are accession numbers and URLs for a project, samples within the project, and sequence datasets from a given sample.</p>
<p><strong>External unique ID and URL:</strong> For NCBI GenBank this would be the accession number for a collection. For most sequence and genomic datasets an access URL would include an accession number (e.g.&nbsp;<a href="https://www.ncbi.nlm.nih.gov/nuccore/AY741555">https://www.ncbi.nlm.nih.gov/nuccore/AY741555</a>). Referring to a range of accession numbers, may involve providing a search URL that will return the desired list, e.g.&nbsp;(<a href="https://www.ncbi.nlm.nih.gov/popset/?term=AY741555">https://www.ncbi.nlm.nih.gov/popset/?term=AY741555</a>). The recommendation is to link to the widest level of sequence or genomic granularity that is useful to interpret data being archived in the new dataset. The following are suggestions for additional contextual columns in the inventory table. This information is generally associated with the data in the genomics repository and should only be duplicated if deemed useful for reuse, or if missing in the original data.</p>
<ul>
<li><strong>Sequencing method</strong>: the name of the sequencing method used; e.g., Sanger, pyrosequencing, ABI-solid. This attribute is used in MIxS templates, where it is called seq_meth.</li>
<li><strong>Environment (biome, feature, or material) descriptors</strong>: These are descriptors of the environmental context and are standardized by the genomics community in the <a href="https://press3.mcs.anl.gov/gensc/mixs/">MixS templates</a> and <a href="www.environmentontology.org/Browse-EnvO">EnvO</a>.</li>
<li><strong>Taxon description</strong>: If applicable, e.g., Binomial name, or taxonomic group</li>
</ul>
<p>Data packages of metadata and inventory tables will aid in discovering genomic data within an ecological data repository (EDI) and will aid in clarifying the context in which they were collected. Most use cases, however, employ this inventory table to link specific genetic data to derived data. Such products frequently are community or population metrics where species, OTUs or traits have been determined from the sequence data.</p>
</section>
</section>
<section id="example-data-packages-in-edi-2" class="level3" data-number="3.6.4">
<h3 data-number="3.6.4" class="anchored" data-anchor-id="example-data-packages-in-edi-2"><span class="header-section-number">3.6.4</span> Example data packages in EDI</h3>
<p>Each of the EDI data packages below are linked to data in outside repositories. Some contain data inventory tables (as dataTable entities) that link to the datasets held in outside repositories and are described in the EML metadata. The EML abstract and methods elements in each give detailed access and citation instructions.</p>
<p>Table 3: Linked data packages at EDI that provide examples of the best practices in this document.</p>
<table>
<tbody><tr>
<td>
<strong>Title</strong>
</td>
<td>
<strong>Description</strong>
</td>
<td>
<strong>EDI packageID</strong>
</td>
</tr>
<tr>
<td>
<em>Mass and energy fluxes from the US-Jo2 AmeriFlux eddy covariance tower in Tromble Weir experimental watershed at the Jornada Basin LTER site, 2010-ongoing</em>
</td>
<td>
This data package links to eddy covariance data from a Jornada Basin LTER tower. The data are held at the AmeriFlux data repository (<a href="https://ameriflux.lbl.gov/">https://ameriflux.lbl.gov</a>)
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?scope=knb-lter-jrn&amp;identifier=210338005">knb-lter-jrn.210338005</a>
</td>
</tr>
<tr>
<td>
<em>Catalog of GenBank sequence read archive (SRA) entries of 16S and 18S rRNA genes from bacterial and protistan planktonic communities along the Eastern Beaufort Sea coast, North Slope, Alaska, 2011-2013</em>
</td>
<td>
Data inventory of runs, samples, and experiments held at GenBank.
</td>
<td>
<a href="https://doi.org/10.6073/pasta/0e4d75453560ab5c085c9b547be68731">knb-lter-ble.10</a>
</td>
</tr>
<tr>
<td>
<em>Correlation of native and exotic species richness: a global meta-analysis finds no invasion paradox across scales</em>
</td>
<td>
This data package re-publishes data held in a package in Dryad. The metadata has been substantially enriched relative to the original dataset.
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?packageid=edi.548.1">edi.548.1</a>
</td>
</tr>
<tr>
<td>
<em>Vascular Flora of the Harvard Farm at Harvard Forest since 2014</em>
</td>
<td>
This data package includes an inventory table with information on voucher specimens held in the Harvard Herbarium.
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?packageid=knb-lter-hfr.236.3">knb-lter-hfr.236.3</a>
</td>
</tr>
<tr>
<td>
<em>Biological responses to landscape change in the McMurdo Dry Valleys, Antarctica</em>
</td>
<td>
This data package links to genomic data in NCBI, and includes additional data from biogeochemical analyses performed on each sample.
</td>
<td>
<a href="https://portal.edirepository.org/nis/mapbrowse?packageid=knb-lter-mcm.262.1">knb-lter-mcm.262.1</a>
</td>
</tr>
</tbody></table>
</section>
<section id="appendix-tips-and-repository-information" class="level3" data-number="3.6.5">
<h3 data-number="3.6.5" class="anchored" data-anchor-id="appendix-tips-and-repository-information"><span class="header-section-number">3.6.5</span> Appendix: Tips and repository information</h3>
<p>This section aggregates information helpful at the time this document was written, particularly regarding nucleotide sequence and genomic data repositories in widespread use at this time. Given the rapid rate of change in the field, this info may fall out of date quickly.</p>
<section id="sequence-and-genomic-repository-information" class="level4" data-number="3.6.5.1">
<h4 data-number="3.6.5.1" class="anchored" data-anchor-id="sequence-and-genomic-repository-information"><span class="header-section-number">3.6.5.1</span> Sequence and genomic repository information</h4>
<p>It is generally preferable that sequencing and genomic data are archived in community repositories that are specialized for their data type, rather than in a generalist repository such as the <a href="https://edirepository.org/">Environmental Data Initiative</a> (EDI). There are many such specialized repositories; a fairly comprehensive listing is provided by the journal <em><a href="http://nar.oxfordjournals.org/">Nucleic Acids Research</a></em> (summarized on <a href="https://en.wikipedia.org/wiki/List_of_biological_databases">this page</a>). Metadata standards and collaborative structures among these repositories are governed by the <a href="http://www.insdc.org/">International Nucleotide Sequence Database Collaboration</a> (INSDC, more guidance <a href="https://www.nature.com/sdata/policies/repositories#nuc">here</a>).Often these repositories provide or are accessible to specialized tools for searching, accessing, and analyzing the data (e.g., BLAST, MG-RAST). Furthermore, some products derived from sequence or genomic data are best archived in another specialized repository (e.g., metagenome-assembled genomes, or MAGs). As a general rule, these specialized repositories assign unique identifiers to projects, samples, and/or single sequences (often referred to as accession numbers) that can be used to locate sequences or genomic data. Note that each repository may have its own mechanism for reverse linking to related data held in another repository (such as EDI), and these mechanisms are beyond the scope of this document.</p>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/search/">NCBI Databases</a> - list of various databases with search capabilities. See also <a href="https://www.ncbi.nlm.nih.gov/genbank/submit/">How to submit data to GenBank</a>.</li>
<li><a href="https://www.ncbi.nlm.nih.gov/Sequin/acc.html">NCBI Accession Number prefixes</a> - Explanation of accession number prefix codes.</li>
<li><a href="http://www.ddbj.nig.ac.jp/">DNA DataBank of Japan (DDBJ)</a> - list of various databases with search capabilities. See also <a href="https://www.ddbj.nig.ac.jp/submission-e.html">Submissions</a>.</li>
<li><a href="http://www.ebi.ac.uk/ena/">European Nucleotide Archive (ENA)</a> - list of various databases with search capabilities. See also <a href="https://www.ebi.ac.uk/ena/submit">Submit and update</a>.</li>
<li><a href="https://img.jgi.doe.gov/">Integrated Microbial Genomes &amp; Microbiomes (IMG/M) system</a> from the Joint Genome Institute</li>
<li><a href="http://www.mg-rast.org/mgmain.html?mgpage=downloadintro">MG-RAST</a> (technically an analysis pipeline not a primary repository, but replicates to primary repositories) Replicates to the European Bioinformatics Institute (EMBL-EBI), which in turn replicates to the NCBI Sequence Read Archive (such that data submitted on MG-RAST will automatically appear on all three).</li>
<li><a href="http://www.barcodinglife.com/">Barcode of Life DataSystems (BOLD)</a> DNA barcoding is a taxonomic method that uses one or more standardized short genetic markers in an organism’s DNA to identify it as belonging to a particular species. Through this method unknown DNA samples are identified to registered species based on comparison to a reference library. The Centre for Biodiversity Genomics in Canada maintains the BOLD public data portal, a cloud-based data storage and analysis platform.</li>
</ul>
</section>
<section id="tips-for-locating-metadata-in-sequence-and-genomic-data-repositories" class="level4" data-number="3.6.5.2">
<h4 data-number="3.6.5.2" class="anchored" data-anchor-id="tips-for-locating-metadata-in-sequence-and-genomic-data-repositories"><span class="header-section-number">3.6.5.2</span> Tips for locating metadata in sequence and genomic data repositories</h4>
<p>Where information for populating metadata in EML has not been supplied directly to the IM from the research group, metadata that investigators provided when submitting data may be found in the genomics repository.</p>
<ul>
<li>For data in NCBI, go to the <a href="https://www.ncbi.nlm.nih.gov/search/">NCBI website</a> and search using the accession number. Or search by accession number in a specific NCBI Database, for example Genes PopSet (the PopSet database is a collection of related DNA sequences derived from population, phylogenetic, mutation and ecosystem studies that have been submitted to NCBI).</li>
<li>For sequences submitted to the <a href="https://www.ncbi.nlm.nih.gov/sra">NCBI Sequence Read Archive</a>, there are some easily accessible online tools for generating tables of linked sequence data and their metadata. For an example, go to the example dataset at <a href="https://www.ncbi.nlm.nih.gov/bioproject/305753">https://www.ncbi.nlm.nih.gbov/bioproject/305753</a>, and click the number next to <strong>SRA Experiments</strong> to see a list of all experiments. Then click <strong>Send results to Run selector</strong> to see a table summarizing geolocations and associated metadata which could be archived at EDI or used to extract metadata for EML preparation.</li>
<li>A full Data Carpentry tutorial on accessing data on the NCBI SRA database can be found here: <a href="https://datacarpentry.org/organization-genomics/03-ncbi-sra/index.html">Examining Data on the NCBI SRA Database</a></li>
<li><a href="https://www.bco-dmo.org/contributing-sequence-accession-numbers">BCO-DMO examples</a> for contributing sequence accession numbers.</li>
</ul>
</section>
<section id="darwin-core-standard-for-sequence-data" class="level4" data-number="3.6.5.3">
<h4 data-number="3.6.5.3" class="anchored" data-anchor-id="darwin-core-standard-for-sequence-data"><span class="header-section-number">3.6.5.3</span> Darwin Core standard for sequence data</h4>
<p>For sequence data to conform with the Darwin Core standard, a column header ‘associatedSequences’ (<a href="https://dwc.tdwg.org/terms/#dwc:associatedSequences">https://dwc.tdwg.org/terms/#dwc:associatedSequences</a>) may be used in the inventory table populated with a unique identifier (or list of identifiers) for the sequence data (e.g., SNLBE002-17, a sequence in Barcode Of Life Data system, aka BOLD) or full URL (e.g., <a href="http://www.boldsystems.org/index.php/Public_RecordView?processid=SNLBE002-17">http://www.boldsystems.org/index.php/Public_RecordView?processid=SNLBE002-17</a>).</p>
</section>
</section>
</section>
<section id="large-data-sets" class="level2" data-number="3.7">
<h2 data-number="3.7" class="anchored" data-anchor-id="large-data-sets"><span class="header-section-number">3.7</span> Large Data Sets</h2>
<p>Contributors: Margaret O’Brien, Corinna Gries, Mark Servilla</p>
<section id="introduction-6" class="level3" data-number="3.7.1">
<h3 data-number="3.7.1" class="anchored" data-anchor-id="introduction-6"><span class="header-section-number">3.7.1</span> Introduction</h3>
<p>Data entities are kept offline when they are too large to be handled easily by the HTTP protocol, are expected to be rarely requested, and can be mailed on an external drive. If you suspect your data fall into this category, contact EDI for advice (support@edirepository.org). Below are recommendations for the EDI repository’s handling of data packages that have an offline component.</p>
<section id="background" class="level4" data-number="3.7.1.1">
<h4 data-number="3.7.1.1" class="anchored" data-anchor-id="background"><span class="header-section-number">3.7.1.1</span> Background</h4>
<p>Standard practice is to handle data entities (both upload and download) via the HTTP protocol, using a URL. However, for very large datasets HTTP can fail due to physical limits. The limit for “too large” is somewhat subjective; EDI’s current limit for datasets that are “too large for HTTP” is 100GB (all data and metadata).</p>
</section>
</section>
<section id="recommendations-for-data-packages-6" class="level3" data-number="3.7.2">
<h3 data-number="3.7.2" class="anchored" data-anchor-id="recommendations-for-data-packages-6"><span class="header-section-number">3.7.2</span> Recommendations for data packages</h3>
<section id="physical-storage" class="level4" data-number="3.7.2.1">
<h4 data-number="3.7.2.1" class="anchored" data-anchor-id="physical-storage"><span class="header-section-number">3.7.2.1</span> Physical Storage</h4>
<ul>
<li><p>The use of a Solid-state Drive (SSD) is strongly recommended for all offline data storage. The SSD should be formatted using one of the following file systems: 1) exFAT, 2) NTFS, or 3) ext4. Each of these file systems can accommodate individual file sizes greater than 1TB.</p></li>
<li><p>Add data to external drive in native (non-compressed, non-tarred, non-zipped) format, deliver to EDI (e.g., by physical mail).</p></li>
<li><p>EDI will store three copies, one external hard drive each in New Mexico and in Wisconsin, one copy in general EDI backup cloud storage.</p></li>
<li><p>Please mail one copy each to:</p></li>
</ul>
<p>Attn: Mark Servilla UNM Biology, Castetter Hall 1480 MSC03-2020, 219 Yale Blvd NE Albuquerque, NM 87131-0001</p>
<p>Attn: Corinna Gries University of Wisconsin Center for Limnology 680 North Park Street Madison WI 53706-1413</p>
</section>
<section id="access" class="level4" data-number="3.7.2.2">
<h4 data-number="3.7.2.2" class="anchored" data-anchor-id="access"><span class="header-section-number">3.7.2.2</span> Access</h4>
<p>To receive offline data, send a request to support@edirepository.org listing the data package(s) of interest and we’ll determine the best mode of delivery. If technically feasible, we will stage data on an internet accessible host and will send you a URL when the data are ready for download. Otherwise, data will be sent on physical media. To receive data on physical media, you will be required to send a storage device of adequate volume to:</p>
<p>Attn: Mark Servilla UNM Biology, Castetter Hall 1480 MSC03-2020 219 Yale Blvd NE Albuquerque, NM 87131-0001</p>
<p>We will utilize a delivery service of your choice, however, you will be responsible for all shipping arrangements and fees, including pre-arranging return delivery.</p>
</section>
<section id="data-package" class="level4" data-number="3.7.2.3">
<h4 data-number="3.7.2.3" class="anchored" data-anchor-id="data-package"><span class="header-section-number">3.7.2.3</span> Data package</h4>
<ul>
<li>The external hard drive should contain at least two entities: the data (which will be offline) and an inventory or manifest that describe the contents of the external hard drive.</li>
<li>Content of the manifest (inventory of holdings) would be dictated by the type of data entity. The <strong>manifest will be available as an online entity</strong> (through the EDI Data Portal) so that potential requestors can evaluate the offline resource before requesting it.</li>
<li>Suggested columns are:
<ul>
<li>Filename(s)</li>
<li>Format (netCDF, tabular csv, etc.)</li>
<li>Start_datetime</li>
<li>End_datetime</li>
<li>Location_lat</li>
<li>Location_lon</li>
<li>(other params the PIs may feel are essential)</li>
<li>Checksum</li>
</ul></li>
</ul>
</section>
<section id="package-metadata" class="level4" data-number="3.7.2.4">
<h4 data-number="3.7.2.4" class="anchored" data-anchor-id="package-metadata"><span class="header-section-number">3.7.2.4</span> Package Metadata</h4>
<p>(in EDI metadata template and converted to EML - generally, as for any data package)</p>
<ul>
<li>Abstract: describe the collection generally. If individual files require specific software to read, provide the name of that software.</li>
<li>Creators</li>
<li>Contact - The first listed contact is responsible for sending out copies as requested and is typically the EDI repository manager:</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="sourceCode xml code-with-copy"><code class="sourceCode xml"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>&lt;<span class="kw">contact</span>&gt;</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  &lt;<span class="kw">organizationName</span>&gt;Environmental Data Initiative&lt;/<span class="kw">organizationName</span>&gt;</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  &lt;<span class="kw">electronicMailAddress</span>&gt;support@edirepository.org&lt;/<span class="kw">electronicMailAddress</span>&gt;</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>&lt;/<span class="kw">contact</span>&gt;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li>Methods - detailed collection/generation methods for the offline data entities. Detailed information for re-using the data. (May instead be included in the manifest table if different for different offline files.)</li>
<li>Data Entities
<ul>
<li>Offline Entity:
<ul>
<li>Describe as you would for an online resource. Restate the software needed to read the individual files if this is important to a user. See <a href="#table_1">Table 1</a> and <a href="#sample_xml_offline">Sample XML</a>.</li>
</ul></li>
<li>Manifest (inventory of the offline holdings)
<ul>
<li>Column descriptions as for any data table</li>
</ul></li>
</ul></li>
</ul>
</section>
<section id="eml" class="level4" data-number="3.7.2.5">
<h4 data-number="3.7.2.5" class="anchored" data-anchor-id="eml"><span class="header-section-number">3.7.2.5</span> EML</h4>
<p>In addition to basic resource-level metadata, at least two entities should be described:</p>
<ul>
<li>Manifest (inventory) should be a tableEntity: will be the online entity and described as all</li>
<li>Offline entity:
<ul>
<li>Fill out high-level fields as for an online resource. Restate the software needed to read the individual files if this is important to a user.</li>
<li>Distribution node will be <code>offline</code> (See Table 1, code block)</li>
</ul></li>
</ul>
<section id="table_1" class="level5" data-number="3.7.2.5.1">
<h5 data-number="3.7.2.5.1" class="anchored" data-anchor-id="table_1"><span class="header-section-number">3.7.2.5.1</span> Table 1. Three required fields for an offline distribution</h5>
<table>
<tbody><tr>
<td>
physical/objectName
</td>
<td>
As for any entity, this is the name of the file or data object
</td>
</tr>
<tr>
<td>
dataFormat/ExternallyDefinedFormat/formatName
</td>
<td>
The name of the format the data object is in. If there is a special compression applied, list it here.
</td>
</tr>
<tr>
<td>
distribution/offline/mediumName
</td>
<td>
Instead of a data URL, you will have an offline distribution node. The name of almost all offline media is “external drive”, because that is how you will deliver the data to a requestor.
</td>
</tr>
</tbody></table>
</section>
<section id="sample_xml_offline" class="level5" data-number="3.7.2.5.2">
<h5 data-number="3.7.2.5.2" class="anchored" data-anchor-id="sample_xml_offline"><span class="header-section-number">3.7.2.5.2</span> Sample XML, offline entity</h5>
<div class="sourceCode" id="cb12"><pre class="sourceCode xml code-with-copy"><code class="sourceCode xml"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>&lt;<span class="kw">physical</span>&gt;</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  &lt;<span class="kw">objectName</span>&gt;mainl_2005acc.zip&lt;/<span class="kw">objectName</span>&gt;</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  &lt;<span class="kw">dataFormat</span>&gt;</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    &lt;<span class="kw">externallyDefinedFormat</span>&gt;</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">formatName</span>&gt;netCDF file&lt;/<span class="kw">formatName</span>&gt;</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    &lt;/<span class="kw">externallyDefinedFormat</span>&gt;</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  &lt;/<span class="kw">dataFormat</span>&gt;</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  &lt;<span class="kw">distribution</span>&gt;</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    &lt;<span class="kw">offline</span>&gt;</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>      &lt;<span class="kw">mediumName</span>&gt;External drive&lt;/<span class="kw">mediumName</span>&gt;</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>    &lt;/<span class="kw">offline</span>&gt;</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  &lt;/<span class="kw">distribution</span>&gt;</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>&lt;/<span class="kw">physical</span>&gt;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
</section>
<section id="references-1" class="level3" data-number="3.7.3">
<h3 data-number="3.7.3" class="anchored" data-anchor-id="references-1"><span class="header-section-number">3.7.3</span> References</h3>
<section id="eml-documentation" class="level4" data-number="3.7.3.1">
<h4 data-number="3.7.3.1" class="anchored" data-anchor-id="eml-documentation"><span class="header-section-number">3.7.3.1</span> EML documentation</h4>
<p><a href="https://eml.ecoinformatics.org/schema/index.html">https://eml.ecoinformatics.org/schema/index.html</a></p>
<p>Look for the PhysicalDistributionType</p>
</section>
</section>
<section id="potential-issues" class="level3" data-number="3.7.4">
<h3 data-number="3.7.4" class="anchored" data-anchor-id="potential-issues"><span class="header-section-number">3.7.4</span> Potential Issues</h3>
<ul>
<li>SSD formatting (eventually, whatever we use, it will become unusable).</li>
<li>Even with cloud storage, eventually a binary format will become unusable.</li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/ediorg\.github\.io\/dataset_preparation_guides");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../guide-special-cases/preface.html" class="pagination-link" aria-label="Data Package Design for Special Cases">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Data Package Design for Special Cases</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../guide-domain-specific/preface.html" class="pagination-link" aria-label="Scientific domain specific metadata guide">
        <span class="nav-page-text">Scientific domain specific metadata guide</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ediorg/dataset_preparation_guides/edit/main/guide-special-cases/dataset-design-special-cases.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ediorg/dataset_preparation_guides/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>